// @polar-sh/sdk@0.18.0/models/components/benefitgrantwebhook.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGrantWebhook$ = exports.BenefitGrantWebhook$outboundSchema = exports.BenefitGrantWebhook$inboundSchema = exports.PreviousProperties$ = exports.PreviousProperties$outboundSchema = exports.PreviousProperties$inboundSchema = exports.BenefitGrantWebhookProperties$ = exports.BenefitGrantWebhookProperties$outboundSchema = exports.BenefitGrantWebhookProperties$inboundSchema = void 0;
exports.benefitGrantWebhookPropertiesToJSON = benefitGrantWebhookPropertiesToJSON;
exports.benefitGrantWebhookPropertiesFromJSON = benefitGrantWebhookPropertiesFromJSON;
exports.previousPropertiesToJSON = previousPropertiesToJSON;
exports.previousPropertiesFromJSON = previousPropertiesFromJSON;
exports.benefitGrantWebhookToJSON = benefitGrantWebhookToJSON;
exports.benefitGrantWebhookFromJSON = benefitGrantWebhookFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefit_js_1 = require("./benefit.js");
const benefitgrantadsproperties_js_1 = require("./benefitgrantadsproperties.js");
const benefitgrantcustomproperties_js_1 = require("./benefitgrantcustomproperties.js");
const benefitgrantdiscordproperties_js_1 = require("./benefitgrantdiscordproperties.js");
const benefitgrantdownloadablesproperties_js_1 = require("./benefitgrantdownloadablesproperties.js");
const benefitgrantgithubrepositoryproperties_js_1 = require("./benefitgrantgithubrepositoryproperties.js");
const benefitgrantlicensekeysproperties_js_1 = require("./benefitgrantlicensekeysproperties.js");
/** @internal */
exports.BenefitGrantWebhookProperties$inboundSchema = z.union([
    benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$inboundSchema,
    benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$inboundSchema,
    benefitgrantadsproperties_js_1.BenefitGrantAdsProperties$inboundSchema,
    benefitgrantlicensekeysproperties_js_1.BenefitGrantLicenseKeysProperties$inboundSchema,
    benefitgrantdiscordproperties_js_1.BenefitGrantDiscordProperties$inboundSchema,
    benefitgrantgithubrepositoryproperties_js_1.BenefitGrantGitHubRepositoryProperties$inboundSchema,
]);
/** @internal */
exports.BenefitGrantWebhookProperties$outboundSchema = z.union([
    benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$outboundSchema,
    benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$outboundSchema,
    benefitgrantadsproperties_js_1.BenefitGrantAdsProperties$outboundSchema,
    benefitgrantlicensekeysproperties_js_1.BenefitGrantLicenseKeysProperties$outboundSchema,
    benefitgrantdiscordproperties_js_1.BenefitGrantDiscordProperties$outboundSchema,
    benefitgrantgithubrepositoryproperties_js_1.BenefitGrantGitHubRepositoryProperties$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGrantWebhookProperties$;
(function (BenefitGrantWebhookProperties$) {
    /** @deprecated use `BenefitGrantWebhookProperties$inboundSchema` instead. */
    BenefitGrantWebhookProperties$.inboundSchema = exports.BenefitGrantWebhookProperties$inboundSchema;
    /** @deprecated use `BenefitGrantWebhookProperties$outboundSchema` instead. */
    BenefitGrantWebhookProperties$.outboundSchema = exports.BenefitGrantWebhookProperties$outboundSchema;
})(BenefitGrantWebhookProperties$ || (exports.BenefitGrantWebhookProperties$ = BenefitGrantWebhookProperties$ = {}));
function benefitGrantWebhookPropertiesToJSON(benefitGrantWebhookProperties) {
    return JSON.stringify(exports.BenefitGrantWebhookProperties$outboundSchema.parse(benefitGrantWebhookProperties));
}
function benefitGrantWebhookPropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGrantWebhookProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantWebhookProperties' from JSON`);
}
/** @internal */
exports.PreviousProperties$inboundSchema = z.union([
    benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$inboundSchema,
    benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$inboundSchema,
    benefitgrantadsproperties_js_1.BenefitGrantAdsProperties$inboundSchema,
    benefitgrantlicensekeysproperties_js_1.BenefitGrantLicenseKeysProperties$inboundSchema,
    benefitgrantdiscordproperties_js_1.BenefitGrantDiscordProperties$inboundSchema,
    benefitgrantgithubrepositoryproperties_js_1.BenefitGrantGitHubRepositoryProperties$inboundSchema,
]);
/** @internal */
exports.PreviousProperties$outboundSchema = z.union([
    benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$outboundSchema,
    benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$outboundSchema,
    benefitgrantadsproperties_js_1.BenefitGrantAdsProperties$outboundSchema,
    benefitgrantlicensekeysproperties_js_1.BenefitGrantLicenseKeysProperties$outboundSchema,
    benefitgrantdiscordproperties_js_1.BenefitGrantDiscordProperties$outboundSchema,
    benefitgrantgithubrepositoryproperties_js_1.BenefitGrantGitHubRepositoryProperties$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PreviousProperties$;
(function (PreviousProperties$) {
    /** @deprecated use `PreviousProperties$inboundSchema` instead. */
    PreviousProperties$.inboundSchema = exports.PreviousProperties$inboundSchema;
    /** @deprecated use `PreviousProperties$outboundSchema` instead. */
    PreviousProperties$.outboundSchema = exports.PreviousProperties$outboundSchema;
})(PreviousProperties$ || (exports.PreviousProperties$ = PreviousProperties$ = {}));
function previousPropertiesToJSON(previousProperties) {
    return JSON.stringify(exports.PreviousProperties$outboundSchema.parse(previousProperties));
}
function previousPropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PreviousProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PreviousProperties' from JSON`);
}
/** @internal */
exports.BenefitGrantWebhook$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    granted_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    is_granted: z.boolean(),
    revoked_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    is_revoked: z.boolean(),
    subscription_id: z.nullable(z.string()),
    order_id: z.nullable(z.string()),
    user_id: z.string(),
    benefit_id: z.string(),
    properties: z.union([
        benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$inboundSchema,
        benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$inboundSchema,
        benefitgrantadsproperties_js_1.BenefitGrantAdsProperties$inboundSchema,
        benefitgrantlicensekeysproperties_js_1.BenefitGrantLicenseKeysProperties$inboundSchema,
        benefitgrantdiscordproperties_js_1.BenefitGrantDiscordProperties$inboundSchema,
        benefitgrantgithubrepositoryproperties_js_1.BenefitGrantGitHubRepositoryProperties$inboundSchema,
    ]),
    benefit: benefit_js_1.Benefit$inboundSchema,
    previous_properties: z.nullable(z.union([
        benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$inboundSchema,
        benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$inboundSchema,
        benefitgrantadsproperties_js_1.BenefitGrantAdsProperties$inboundSchema,
        benefitgrantlicensekeysproperties_js_1.BenefitGrantLicenseKeysProperties$inboundSchema,
        benefitgrantdiscordproperties_js_1.BenefitGrantDiscordProperties$inboundSchema,
        benefitgrantgithubrepositoryproperties_js_1.BenefitGrantGitHubRepositoryProperties$inboundSchema,
    ])).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "granted_at": "grantedAt",
        "is_granted": "isGranted",
        "revoked_at": "revokedAt",
        "is_revoked": "isRevoked",
        "subscription_id": "subscriptionId",
        "order_id": "orderId",
        "user_id": "userId",
        "benefit_id": "benefitId",
        "previous_properties": "previousProperties",
    });
});
/** @internal */
exports.BenefitGrantWebhook$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    grantedAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    isGranted: z.boolean(),
    revokedAt: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    isRevoked: z.boolean(),
    subscriptionId: z.nullable(z.string()),
    orderId: z.nullable(z.string()),
    userId: z.string(),
    benefitId: z.string(),
    properties: z.union([
        benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$outboundSchema,
        benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$outboundSchema,
        benefitgrantadsproperties_js_1.BenefitGrantAdsProperties$outboundSchema,
        benefitgrantlicensekeysproperties_js_1.BenefitGrantLicenseKeysProperties$outboundSchema,
        benefitgrantdiscordproperties_js_1.BenefitGrantDiscordProperties$outboundSchema,
        benefitgrantgithubrepositoryproperties_js_1.BenefitGrantGitHubRepositoryProperties$outboundSchema,
    ]),
    benefit: benefit_js_1.Benefit$outboundSchema,
    previousProperties: z.nullable(z.union([
        benefitgrantcustomproperties_js_1.BenefitGrantCustomProperties$outboundSchema,
        benefitgrantdownloadablesproperties_js_1.BenefitGrantDownloadablesProperties$outboundSchema,
        benefitgrantadsproperties_js_1.BenefitGrantAdsProperties$outboundSchema,
        benefitgrantlicensekeysproperties_js_1.BenefitGrantLicenseKeysProperties$outboundSchema,
        benefitgrantdiscordproperties_js_1.BenefitGrantDiscordProperties$outboundSchema,
        benefitgrantgithubrepositoryproperties_js_1.BenefitGrantGitHubRepositoryProperties$outboundSchema,
    ])).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        grantedAt: "granted_at",
        isGranted: "is_granted",
        revokedAt: "revoked_at",
        isRevoked: "is_revoked",
        subscriptionId: "subscription_id",
        orderId: "order_id",
        userId: "user_id",
        benefitId: "benefit_id",
        previousProperties: "previous_properties",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGrantWebhook$;
(function (BenefitGrantWebhook$) {
    /** @deprecated use `BenefitGrantWebhook$inboundSchema` instead. */
    BenefitGrantWebhook$.inboundSchema = exports.BenefitGrantWebhook$inboundSchema;
    /** @deprecated use `BenefitGrantWebhook$outboundSchema` instead. */
    BenefitGrantWebhook$.outboundSchema = exports.BenefitGrantWebhook$outboundSchema;
})(BenefitGrantWebhook$ || (exports.BenefitGrantWebhook$ = BenefitGrantWebhook$ = {}));
function benefitGrantWebhookToJSON(benefitGrantWebhook) {
    return JSON.stringify(exports.BenefitGrantWebhook$outboundSchema.parse(benefitGrantWebhook));
}
function benefitGrantWebhookFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGrantWebhook$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantWebhook' from JSON`);
}
