// @polar-sh/sdk@0.18.0/models/components/benefitdownloadablescreateproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitDownloadablesCreateProperties$ = exports.BenefitDownloadablesCreateProperties$outboundSchema = exports.BenefitDownloadablesCreateProperties$inboundSchema = void 0;
exports.benefitDownloadablesCreatePropertiesToJSON = benefitDownloadablesCreatePropertiesToJSON;
exports.benefitDownloadablesCreatePropertiesFromJSON = benefitDownloadablesCreatePropertiesFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.BenefitDownloadablesCreateProperties$inboundSchema = z.object({
    archived: z.record(z.boolean()).optional(),
    files: z.array(z.string()),
});
/** @internal */
exports.BenefitDownloadablesCreateProperties$outboundSchema = z.object({
    archived: z.record(z.boolean()).optional(),
    files: z.array(z.string()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitDownloadablesCreateProperties$;
(function (BenefitDownloadablesCreateProperties$) {
    /** @deprecated use `BenefitDownloadablesCreateProperties$inboundSchema` instead. */
    BenefitDownloadablesCreateProperties$.inboundSchema = exports.BenefitDownloadablesCreateProperties$inboundSchema;
    /** @deprecated use `BenefitDownloadablesCreateProperties$outboundSchema` instead. */
    BenefitDownloadablesCreateProperties$.outboundSchema = exports.BenefitDownloadablesCreateProperties$outboundSchema;
})(BenefitDownloadablesCreateProperties$ || (exports.BenefitDownloadablesCreateProperties$ = BenefitDownloadablesCreateProperties$ = {}));
function benefitDownloadablesCreatePropertiesToJSON(benefitDownloadablesCreateProperties) {
    return JSON.stringify(exports.BenefitDownloadablesCreateProperties$outboundSchema.parse(benefitDownloadablesCreateProperties));
}
function benefitDownloadablesCreatePropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitDownloadablesCreateProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitDownloadablesCreateProperties' from JSON`);
}
