// @polar-sh/sdk@0.18.0/models/components/benefitadssubscriber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitAdsSubscriber$ = exports.BenefitAdsSubscriber$outboundSchema = exports.BenefitAdsSubscriber$inboundSchema = exports.Type$ = exports.Type$outboundSchema = exports.Type$inboundSchema = exports.Type = void 0;
exports.benefitAdsSubscriberToJSON = benefitAdsSubscriberToJSON;
exports.benefitAdsSubscriberFromJSON = benefitAdsSubscriberFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitadsproperties_js_1 = require("./benefitadsproperties.js");
const benefitgrantads_js_1 = require("./benefitgrantads.js");
const organization_js_1 = require("./organization.js");
exports.Type = {
    Ads: "ads",
};
/** @internal */
exports.Type$inboundSchema = z.nativeEnum(exports.Type);
/** @internal */
exports.Type$outboundSchema = exports.Type$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Type$;
(function (Type$) {
    /** @deprecated use `Type$inboundSchema` instead. */
    Type$.inboundSchema = exports.Type$inboundSchema;
    /** @deprecated use `Type$outboundSchema` instead. */
    Type$.outboundSchema = exports.Type$outboundSchema;
})(Type$ || (exports.Type$ = Type$ = {}));
/** @internal */
exports.BenefitAdsSubscriber$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    type: z.literal("ads").optional(),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organization_id: z.string(),
    grants: z.array(benefitgrantads_js_1.BenefitGrantAds$inboundSchema),
    organization: organization_js_1.Organization$inboundSchema,
    properties: benefitadsproperties_js_1.BenefitAdsProperties$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.BenefitAdsSubscriber$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    type: z.literal("ads").default("ads"),
    description: z.string(),
    selectable: z.boolean(),
    deletable: z.boolean(),
    organizationId: z.string(),
    grants: z.array(benefitgrantads_js_1.BenefitGrantAds$outboundSchema),
    organization: organization_js_1.Organization$outboundSchema,
    properties: benefitadsproperties_js_1.BenefitAdsProperties$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitAdsSubscriber$;
(function (BenefitAdsSubscriber$) {
    /** @deprecated use `BenefitAdsSubscriber$inboundSchema` instead. */
    BenefitAdsSubscriber$.inboundSchema = exports.BenefitAdsSubscriber$inboundSchema;
    /** @deprecated use `BenefitAdsSubscriber$outboundSchema` instead. */
    BenefitAdsSubscriber$.outboundSchema = exports.BenefitAdsSubscriber$outboundSchema;
})(BenefitAdsSubscriber$ || (exports.BenefitAdsSubscriber$ = BenefitAdsSubscriber$ = {}));
function benefitAdsSubscriberToJSON(benefitAdsSubscriber) {
    return JSON.stringify(exports.BenefitAdsSubscriber$outboundSchema.parse(benefitAdsSubscriber));
}
function benefitAdsSubscriberFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitAdsSubscriber$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitAdsSubscriber' from JSON`);
}
