// @polar-sh/sdk@0.18.0/models/components/benefitcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitCreate$ = exports.BenefitCreate$outboundSchema = exports.BenefitCreate$inboundSchema = void 0;
exports.benefitCreateToJSON = benefitCreateToJSON;
exports.benefitCreateFromJSON = benefitCreateFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const benefitadscreate_js_1 = require("./benefitadscreate.js");
const benefitcustomcreate_js_1 = require("./benefitcustomcreate.js");
const benefitdiscordcreate_js_1 = require("./benefitdiscordcreate.js");
const benefitdownloadablescreate_js_1 = require("./benefitdownloadablescreate.js");
const benefitgithubrepositorycreate_js_1 = require("./benefitgithubrepositorycreate.js");
const benefitlicensekeyscreate_js_1 = require("./benefitlicensekeyscreate.js");
/** @internal */
exports.BenefitCreate$inboundSchema = z.union([
    benefitadscreate_js_1.BenefitAdsCreate$inboundSchema.and(z.object({ type: z.literal("ads") }).transform((v) => ({ type: v.type }))),
    benefitdiscordcreate_js_1.BenefitDiscordCreate$inboundSchema.and(z.object({ type: z.literal("discord") }).transform((v) => ({
        type: v.type,
    }))),
    benefitdownloadablescreate_js_1.BenefitDownloadablesCreate$inboundSchema.and(z.object({ type: z.literal("downloadables") }).transform((v) => ({
        type: v.type,
    }))),
    benefitgithubrepositorycreate_js_1.BenefitGitHubRepositoryCreate$inboundSchema.and(z.object({ type: z.literal("github_repository") }).transform((v) => ({
        type: v.type,
    }))),
    benefitlicensekeyscreate_js_1.BenefitLicenseKeysCreate$inboundSchema.and(z.object({ type: z.literal("license_keys") }).transform((v) => ({
        type: v.type,
    }))),
    benefitcustomcreate_js_1.BenefitCustomCreate$inboundSchema.and(z.object({ type: z.literal("custom") }).transform((v) => ({
        type: v.type,
    }))),
]);
/** @internal */
exports.BenefitCreate$outboundSchema = z.union([
    benefitadscreate_js_1.BenefitAdsCreate$outboundSchema.and(z.object({ type: z.literal("ads") }).transform((v) => ({ type: v.type }))),
    benefitdiscordcreate_js_1.BenefitDiscordCreate$outboundSchema.and(z.object({ type: z.literal("discord") }).transform((v) => ({
        type: v.type,
    }))),
    benefitdownloadablescreate_js_1.BenefitDownloadablesCreate$outboundSchema.and(z.object({ type: z.literal("downloadables") }).transform((v) => ({
        type: v.type,
    }))),
    benefitgithubrepositorycreate_js_1.BenefitGitHubRepositoryCreate$outboundSchema.and(z.object({ type: z.literal("github_repository") }).transform((v) => ({
        type: v.type,
    }))),
    benefitlicensekeyscreate_js_1.BenefitLicenseKeysCreate$outboundSchema.and(z.object({ type: z.literal("license_keys") }).transform((v) => ({
        type: v.type,
    }))),
    benefitcustomcreate_js_1.BenefitCustomCreate$outboundSchema.and(z.object({ type: z.literal("custom") }).transform((v) => ({
        type: v.type,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitCreate$;
(function (BenefitCreate$) {
    /** @deprecated use `BenefitCreate$inboundSchema` instead. */
    BenefitCreate$.inboundSchema = exports.BenefitCreate$inboundSchema;
    /** @deprecated use `BenefitCreate$outboundSchema` instead. */
    BenefitCreate$.outboundSchema = exports.BenefitCreate$outboundSchema;
})(BenefitCreate$ || (exports.BenefitCreate$ = BenefitCreate$ = {}));
function benefitCreateToJSON(benefitCreate) {
    return JSON.stringify(exports.BenefitCreate$outboundSchema.parse(benefitCreate));
}
function benefitCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitCreate' from JSON`);
}
