// @polar-sh/sdk@0.18.0/models/components/benefitgrantdiscordproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BenefitGrantDiscordProperties$ = exports.BenefitGrantDiscordProperties$outboundSchema = exports.BenefitGrantDiscordProperties$inboundSchema = void 0;
exports.benefitGrantDiscordPropertiesToJSON = benefitGrantDiscordPropertiesToJSON;
exports.benefitGrantDiscordPropertiesFromJSON = benefitGrantDiscordPropertiesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.BenefitGrantDiscordProperties$inboundSchema = z.object({
    guild_id: z.string().optional(),
    role_id: z.string().optional(),
    account_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "guild_id": "guildId",
        "role_id": "roleId",
        "account_id": "accountId",
    });
});
/** @internal */
exports.BenefitGrantDiscordProperties$outboundSchema = z.object({
    guildId: z.string().optional(),
    roleId: z.string().optional(),
    accountId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        guildId: "guild_id",
        roleId: "role_id",
        accountId: "account_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitGrantDiscordProperties$;
(function (BenefitGrantDiscordProperties$) {
    /** @deprecated use `BenefitGrantDiscordProperties$inboundSchema` instead. */
    BenefitGrantDiscordProperties$.inboundSchema = exports.BenefitGrantDiscordProperties$inboundSchema;
    /** @deprecated use `BenefitGrantDiscordProperties$outboundSchema` instead. */
    BenefitGrantDiscordProperties$.outboundSchema = exports.BenefitGrantDiscordProperties$outboundSchema;
})(BenefitGrantDiscordProperties$ || (exports.BenefitGrantDiscordProperties$ = BenefitGrantDiscordProperties$ = {}));
function benefitGrantDiscordPropertiesToJSON(benefitGrantDiscordProperties) {
    return JSON.stringify(exports.BenefitGrantDiscordProperties$outboundSchema.parse(benefitGrantDiscordProperties));
}
function benefitGrantDiscordPropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitGrantDiscordProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitGrantDiscordProperties' from JSON`);
}
