// @polar-sh/sdk@0.18.0/models/components/checkoutpricecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutPriceCreate$ = exports.CheckoutPriceCreate$outboundSchema = exports.CheckoutPriceCreate$inboundSchema = exports.CheckoutPriceCreatePaymentProcessor$ = exports.CheckoutPriceCreatePaymentProcessor$outboundSchema = exports.CheckoutPriceCreatePaymentProcessor$inboundSchema = exports.CheckoutPriceCreateCustomFieldData$ = exports.CheckoutPriceCreateCustomFieldData$outboundSchema = exports.CheckoutPriceCreateCustomFieldData$inboundSchema = exports.CheckoutPriceCreateMetadata$ = exports.CheckoutPriceCreateMetadata$outboundSchema = exports.CheckoutPriceCreateMetadata$inboundSchema = exports.CheckoutPriceCreatePaymentProcessor = void 0;
exports.checkoutPriceCreateMetadataToJSON = checkoutPriceCreateMetadataToJSON;
exports.checkoutPriceCreateMetadataFromJSON = checkoutPriceCreateMetadataFromJSON;
exports.checkoutPriceCreateCustomFieldDataToJSON = checkoutPriceCreateCustomFieldDataToJSON;
exports.checkoutPriceCreateCustomFieldDataFromJSON = checkoutPriceCreateCustomFieldDataFromJSON;
exports.checkoutPriceCreateToJSON = checkoutPriceCreateToJSON;
exports.checkoutPriceCreateFromJSON = checkoutPriceCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const address_js_1 = require("./address.js");
/**
 * Payment processor to use. Currently only Stripe is supported.
 */
exports.CheckoutPriceCreatePaymentProcessor = {
    Stripe: "stripe",
};
/** @internal */
exports.CheckoutPriceCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
exports.CheckoutPriceCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutPriceCreateMetadata$;
(function (CheckoutPriceCreateMetadata$) {
    /** @deprecated use `CheckoutPriceCreateMetadata$inboundSchema` instead. */
    CheckoutPriceCreateMetadata$.inboundSchema = exports.CheckoutPriceCreateMetadata$inboundSchema;
    /** @deprecated use `CheckoutPriceCreateMetadata$outboundSchema` instead. */
    CheckoutPriceCreateMetadata$.outboundSchema = exports.CheckoutPriceCreateMetadata$outboundSchema;
})(CheckoutPriceCreateMetadata$ || (exports.CheckoutPriceCreateMetadata$ = CheckoutPriceCreateMetadata$ = {}));
function checkoutPriceCreateMetadataToJSON(checkoutPriceCreateMetadata) {
    return JSON.stringify(exports.CheckoutPriceCreateMetadata$outboundSchema.parse(checkoutPriceCreateMetadata));
}
function checkoutPriceCreateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutPriceCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPriceCreateMetadata' from JSON`);
}
/** @internal */
exports.CheckoutPriceCreateCustomFieldData$inboundSchema = z.object({});
/** @internal */
exports.CheckoutPriceCreateCustomFieldData$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutPriceCreateCustomFieldData$;
(function (CheckoutPriceCreateCustomFieldData$) {
    /** @deprecated use `CheckoutPriceCreateCustomFieldData$inboundSchema` instead. */
    CheckoutPriceCreateCustomFieldData$.inboundSchema = exports.CheckoutPriceCreateCustomFieldData$inboundSchema;
    /** @deprecated use `CheckoutPriceCreateCustomFieldData$outboundSchema` instead. */
    CheckoutPriceCreateCustomFieldData$.outboundSchema = exports.CheckoutPriceCreateCustomFieldData$outboundSchema;
})(CheckoutPriceCreateCustomFieldData$ || (exports.CheckoutPriceCreateCustomFieldData$ = CheckoutPriceCreateCustomFieldData$ = {}));
function checkoutPriceCreateCustomFieldDataToJSON(checkoutPriceCreateCustomFieldData) {
    return JSON.stringify(exports.CheckoutPriceCreateCustomFieldData$outboundSchema.parse(checkoutPriceCreateCustomFieldData));
}
function checkoutPriceCreateCustomFieldDataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutPriceCreateCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPriceCreateCustomFieldData' from JSON`);
}
/** @internal */
exports.CheckoutPriceCreatePaymentProcessor$inboundSchema = z.nativeEnum(exports.CheckoutPriceCreatePaymentProcessor);
/** @internal */
exports.CheckoutPriceCreatePaymentProcessor$outboundSchema = exports.CheckoutPriceCreatePaymentProcessor$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutPriceCreatePaymentProcessor$;
(function (CheckoutPriceCreatePaymentProcessor$) {
    /** @deprecated use `CheckoutPriceCreatePaymentProcessor$inboundSchema` instead. */
    CheckoutPriceCreatePaymentProcessor$.inboundSchema = exports.CheckoutPriceCreatePaymentProcessor$inboundSchema;
    /** @deprecated use `CheckoutPriceCreatePaymentProcessor$outboundSchema` instead. */
    CheckoutPriceCreatePaymentProcessor$.outboundSchema = exports.CheckoutPriceCreatePaymentProcessor$outboundSchema;
})(CheckoutPriceCreatePaymentProcessor$ || (exports.CheckoutPriceCreatePaymentProcessor$ = CheckoutPriceCreatePaymentProcessor$ = {}));
/** @internal */
exports.CheckoutPriceCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    custom_field_data: z.lazy(() => exports.CheckoutPriceCreateCustomFieldData$inboundSchema).optional(),
    payment_processor: z.literal("stripe").optional(),
    discount_id: z.nullable(z.string()).optional(),
    allow_discount_codes: z.boolean().default(true),
    amount: z.nullable(z.number().int()).optional(),
    customer_name: z.nullable(z.string()).optional(),
    customer_email: z.nullable(z.string()).optional(),
    customer_ip_address: z.nullable(z.string()).optional(),
    customer_billing_address: z.nullable(address_js_1.Address$inboundSchema).optional(),
    customer_tax_id: z.nullable(z.string()).optional(),
    subscription_id: z.nullable(z.string()).optional(),
    success_url: z.nullable(z.string()).optional(),
    embed_origin: z.nullable(z.string()).optional(),
    product_price_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "custom_field_data": "customFieldData",
        "payment_processor": "paymentProcessor",
        "discount_id": "discountId",
        "allow_discount_codes": "allowDiscountCodes",
        "customer_name": "customerName",
        "customer_email": "customerEmail",
        "customer_ip_address": "customerIpAddress",
        "customer_billing_address": "customerBillingAddress",
        "customer_tax_id": "customerTaxId",
        "subscription_id": "subscriptionId",
        "success_url": "successUrl",
        "embed_origin": "embedOrigin",
        "product_price_id": "productPriceId",
    });
});
/** @internal */
exports.CheckoutPriceCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    customFieldData: z.lazy(() => exports.CheckoutPriceCreateCustomFieldData$outboundSchema).optional(),
    paymentProcessor: z.literal("stripe").default("stripe"),
    discountId: z.nullable(z.string()).optional(),
    allowDiscountCodes: z.boolean().default(true),
    amount: z.nullable(z.number().int()).optional(),
    customerName: z.nullable(z.string()).optional(),
    customerEmail: z.nullable(z.string()).optional(),
    customerIpAddress: z.nullable(z.string()).optional(),
    customerBillingAddress: z.nullable(address_js_1.Address$outboundSchema).optional(),
    customerTaxId: z.nullable(z.string()).optional(),
    subscriptionId: z.nullable(z.string()).optional(),
    successUrl: z.nullable(z.string()).optional(),
    embedOrigin: z.nullable(z.string()).optional(),
    productPriceId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customFieldData: "custom_field_data",
        paymentProcessor: "payment_processor",
        discountId: "discount_id",
        allowDiscountCodes: "allow_discount_codes",
        customerName: "customer_name",
        customerEmail: "customer_email",
        customerIpAddress: "customer_ip_address",
        customerBillingAddress: "customer_billing_address",
        customerTaxId: "customer_tax_id",
        subscriptionId: "subscription_id",
        successUrl: "success_url",
        embedOrigin: "embed_origin",
        productPriceId: "product_price_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutPriceCreate$;
(function (CheckoutPriceCreate$) {
    /** @deprecated use `CheckoutPriceCreate$inboundSchema` instead. */
    CheckoutPriceCreate$.inboundSchema = exports.CheckoutPriceCreate$inboundSchema;
    /** @deprecated use `CheckoutPriceCreate$outboundSchema` instead. */
    CheckoutPriceCreate$.outboundSchema = exports.CheckoutPriceCreate$outboundSchema;
})(CheckoutPriceCreate$ || (exports.CheckoutPriceCreate$ = CheckoutPriceCreate$ = {}));
function checkoutPriceCreateToJSON(checkoutPriceCreate) {
    return JSON.stringify(exports.CheckoutPriceCreate$outboundSchema.parse(checkoutPriceCreate));
}
function checkoutPriceCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutPriceCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutPriceCreate' from JSON`);
}
