// @polar-sh/sdk@0.18.0/models/components/checkoutupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutUpdate$ = exports.CheckoutUpdate$outboundSchema = exports.CheckoutUpdate$inboundSchema = exports.CheckoutUpdateMetadata$ = exports.CheckoutUpdateMetadata$outboundSchema = exports.CheckoutUpdateMetadata$inboundSchema = exports.CustomFieldData$ = exports.CustomFieldData$outboundSchema = exports.CustomFieldData$inboundSchema = void 0;
exports.customFieldDataToJSON = customFieldDataToJSON;
exports.customFieldDataFromJSON = customFieldDataFromJSON;
exports.checkoutUpdateMetadataToJSON = checkoutUpdateMetadataToJSON;
exports.checkoutUpdateMetadataFromJSON = checkoutUpdateMetadataFromJSON;
exports.checkoutUpdateToJSON = checkoutUpdateToJSON;
exports.checkoutUpdateFromJSON = checkoutUpdateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const address_js_1 = require("./address.js");
/** @internal */
exports.CustomFieldData$inboundSchema = z.object({});
/** @internal */
exports.CustomFieldData$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldData$;
(function (CustomFieldData$) {
    /** @deprecated use `CustomFieldData$inboundSchema` instead. */
    CustomFieldData$.inboundSchema = exports.CustomFieldData$inboundSchema;
    /** @deprecated use `CustomFieldData$outboundSchema` instead. */
    CustomFieldData$.outboundSchema = exports.CustomFieldData$outboundSchema;
})(CustomFieldData$ || (exports.CustomFieldData$ = CustomFieldData$ = {}));
function customFieldDataToJSON(customFieldData) {
    return JSON.stringify(exports.CustomFieldData$outboundSchema.parse(customFieldData));
}
function customFieldDataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldData' from JSON`);
}
/** @internal */
exports.CheckoutUpdateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
exports.CheckoutUpdateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutUpdateMetadata$;
(function (CheckoutUpdateMetadata$) {
    /** @deprecated use `CheckoutUpdateMetadata$inboundSchema` instead. */
    CheckoutUpdateMetadata$.inboundSchema = exports.CheckoutUpdateMetadata$inboundSchema;
    /** @deprecated use `CheckoutUpdateMetadata$outboundSchema` instead. */
    CheckoutUpdateMetadata$.outboundSchema = exports.CheckoutUpdateMetadata$outboundSchema;
})(CheckoutUpdateMetadata$ || (exports.CheckoutUpdateMetadata$ = CheckoutUpdateMetadata$ = {}));
function checkoutUpdateMetadataToJSON(checkoutUpdateMetadata) {
    return JSON.stringify(exports.CheckoutUpdateMetadata$outboundSchema.parse(checkoutUpdateMetadata));
}
function checkoutUpdateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutUpdateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutUpdateMetadata' from JSON`);
}
/** @internal */
exports.CheckoutUpdate$inboundSchema = z.object({
    custom_field_data: z.nullable(z.lazy(() => exports.CustomFieldData$inboundSchema))
        .optional(),
    product_price_id: z.nullable(z.string()).optional(),
    amount: z.nullable(z.number().int()).optional(),
    customer_name: z.nullable(z.string()).optional(),
    customer_email: z.nullable(z.string()).optional(),
    customer_billing_address: z.nullable(address_js_1.Address$inboundSchema).optional(),
    customer_tax_id: z.nullable(z.string()).optional(),
    metadata: z.nullable(z.record(z.union([z.string(), z.number().int(), z.boolean()]))).optional(),
    discount_id: z.nullable(z.string()).optional(),
    allow_discount_codes: z.nullable(z.boolean()).optional(),
    customer_ip_address: z.nullable(z.string()).optional(),
    success_url: z.nullable(z.string()).optional(),
    embed_origin: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "custom_field_data": "customFieldData",
        "product_price_id": "productPriceId",
        "customer_name": "customerName",
        "customer_email": "customerEmail",
        "customer_billing_address": "customerBillingAddress",
        "customer_tax_id": "customerTaxId",
        "discount_id": "discountId",
        "allow_discount_codes": "allowDiscountCodes",
        "customer_ip_address": "customerIpAddress",
        "success_url": "successUrl",
        "embed_origin": "embedOrigin",
    });
});
/** @internal */
exports.CheckoutUpdate$outboundSchema = z.object({
    customFieldData: z.nullable(z.lazy(() => exports.CustomFieldData$outboundSchema))
        .optional(),
    productPriceId: z.nullable(z.string()).optional(),
    amount: z.nullable(z.number().int()).optional(),
    customerName: z.nullable(z.string()).optional(),
    customerEmail: z.nullable(z.string()).optional(),
    customerBillingAddress: z.nullable(address_js_1.Address$outboundSchema).optional(),
    customerTaxId: z.nullable(z.string()).optional(),
    metadata: z.nullable(z.record(z.union([z.string(), z.number().int(), z.boolean()]))).optional(),
    discountId: z.nullable(z.string()).optional(),
    allowDiscountCodes: z.nullable(z.boolean()).optional(),
    customerIpAddress: z.nullable(z.string()).optional(),
    successUrl: z.nullable(z.string()).optional(),
    embedOrigin: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customFieldData: "custom_field_data",
        productPriceId: "product_price_id",
        customerName: "customer_name",
        customerEmail: "customer_email",
        customerBillingAddress: "customer_billing_address",
        customerTaxId: "customer_tax_id",
        discountId: "discount_id",
        allowDiscountCodes: "allow_discount_codes",
        customerIpAddress: "customer_ip_address",
        successUrl: "success_url",
        embedOrigin: "embed_origin",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutUpdate$;
(function (CheckoutUpdate$) {
    /** @deprecated use `CheckoutUpdate$inboundSchema` instead. */
    CheckoutUpdate$.inboundSchema = exports.CheckoutUpdate$inboundSchema;
    /** @deprecated use `CheckoutUpdate$outboundSchema` instead. */
    CheckoutUpdate$.outboundSchema = exports.CheckoutUpdate$outboundSchema;
})(CheckoutUpdate$ || (exports.CheckoutUpdate$ = CheckoutUpdate$ = {}));
function checkoutUpdateToJSON(checkoutUpdate) {
    return JSON.stringify(exports.CheckoutUpdate$outboundSchema.parse(checkoutUpdate));
}
function checkoutUpdateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutUpdate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutUpdate' from JSON`);
}
