// @polar-sh/sdk@0.18.0/models/components/checkoutlinkpricecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutLinkPriceCreate$ = exports.CheckoutLinkPriceCreate$outboundSchema = exports.CheckoutLinkPriceCreate$inboundSchema = exports.CheckoutLinkPriceCreatePaymentProcessor$ = exports.CheckoutLinkPriceCreatePaymentProcessor$outboundSchema = exports.CheckoutLinkPriceCreatePaymentProcessor$inboundSchema = exports.CheckoutLinkPriceCreateMetadata$ = exports.CheckoutLinkPriceCreateMetadata$outboundSchema = exports.CheckoutLinkPriceCreateMetadata$inboundSchema = exports.CheckoutLinkPriceCreatePaymentProcessor = void 0;
exports.checkoutLinkPriceCreateMetadataToJSON = checkoutLinkPriceCreateMetadataToJSON;
exports.checkoutLinkPriceCreateMetadataFromJSON = checkoutLinkPriceCreateMetadataFromJSON;
exports.checkoutLinkPriceCreateToJSON = checkoutLinkPriceCreateToJSON;
exports.checkoutLinkPriceCreateFromJSON = checkoutLinkPriceCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * Payment processor to use. Currently only Stripe is supported.
 */
exports.CheckoutLinkPriceCreatePaymentProcessor = {
    Stripe: "stripe",
};
/** @internal */
exports.CheckoutLinkPriceCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
exports.CheckoutLinkPriceCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkPriceCreateMetadata$;
(function (CheckoutLinkPriceCreateMetadata$) {
    /** @deprecated use `CheckoutLinkPriceCreateMetadata$inboundSchema` instead. */
    CheckoutLinkPriceCreateMetadata$.inboundSchema = exports.CheckoutLinkPriceCreateMetadata$inboundSchema;
    /** @deprecated use `CheckoutLinkPriceCreateMetadata$outboundSchema` instead. */
    CheckoutLinkPriceCreateMetadata$.outboundSchema = exports.CheckoutLinkPriceCreateMetadata$outboundSchema;
})(CheckoutLinkPriceCreateMetadata$ || (exports.CheckoutLinkPriceCreateMetadata$ = CheckoutLinkPriceCreateMetadata$ = {}));
function checkoutLinkPriceCreateMetadataToJSON(checkoutLinkPriceCreateMetadata) {
    return JSON.stringify(exports.CheckoutLinkPriceCreateMetadata$outboundSchema.parse(checkoutLinkPriceCreateMetadata));
}
function checkoutLinkPriceCreateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinkPriceCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkPriceCreateMetadata' from JSON`);
}
/** @internal */
exports.CheckoutLinkPriceCreatePaymentProcessor$inboundSchema = z
    .nativeEnum(exports.CheckoutLinkPriceCreatePaymentProcessor);
/** @internal */
exports.CheckoutLinkPriceCreatePaymentProcessor$outboundSchema = exports.CheckoutLinkPriceCreatePaymentProcessor$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkPriceCreatePaymentProcessor$;
(function (CheckoutLinkPriceCreatePaymentProcessor$) {
    /** @deprecated use `CheckoutLinkPriceCreatePaymentProcessor$inboundSchema` instead. */
    CheckoutLinkPriceCreatePaymentProcessor$.inboundSchema = exports.CheckoutLinkPriceCreatePaymentProcessor$inboundSchema;
    /** @deprecated use `CheckoutLinkPriceCreatePaymentProcessor$outboundSchema` instead. */
    CheckoutLinkPriceCreatePaymentProcessor$.outboundSchema = exports.CheckoutLinkPriceCreatePaymentProcessor$outboundSchema;
})(CheckoutLinkPriceCreatePaymentProcessor$ || (exports.CheckoutLinkPriceCreatePaymentProcessor$ = CheckoutLinkPriceCreatePaymentProcessor$ = {}));
/** @internal */
exports.CheckoutLinkPriceCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    payment_processor: z.literal("stripe").optional(),
    label: z.nullable(z.string()).optional(),
    allow_discount_codes: z.boolean().default(true),
    discount_id: z.nullable(z.string()).optional(),
    success_url: z.nullable(z.string()).optional(),
    product_price_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "payment_processor": "paymentProcessor",
        "allow_discount_codes": "allowDiscountCodes",
        "discount_id": "discountId",
        "success_url": "successUrl",
        "product_price_id": "productPriceId",
    });
});
/** @internal */
exports.CheckoutLinkPriceCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    paymentProcessor: z.literal("stripe").default("stripe"),
    label: z.nullable(z.string()).optional(),
    allowDiscountCodes: z.boolean().default(true),
    discountId: z.nullable(z.string()).optional(),
    successUrl: z.nullable(z.string()).optional(),
    productPriceId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        paymentProcessor: "payment_processor",
        allowDiscountCodes: "allow_discount_codes",
        discountId: "discount_id",
        successUrl: "success_url",
        productPriceId: "product_price_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkPriceCreate$;
(function (CheckoutLinkPriceCreate$) {
    /** @deprecated use `CheckoutLinkPriceCreate$inboundSchema` instead. */
    CheckoutLinkPriceCreate$.inboundSchema = exports.CheckoutLinkPriceCreate$inboundSchema;
    /** @deprecated use `CheckoutLinkPriceCreate$outboundSchema` instead. */
    CheckoutLinkPriceCreate$.outboundSchema = exports.CheckoutLinkPriceCreate$outboundSchema;
})(CheckoutLinkPriceCreate$ || (exports.CheckoutLinkPriceCreate$ = CheckoutLinkPriceCreate$ = {}));
function checkoutLinkPriceCreateToJSON(checkoutLinkPriceCreate) {
    return JSON.stringify(exports.CheckoutLinkPriceCreate$outboundSchema.parse(checkoutLinkPriceCreate));
}
function checkoutLinkPriceCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinkPriceCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkPriceCreate' from JSON`);
}
