// @polar-sh/sdk@0.18.0/models/components/checkoutlink.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutLink$ = exports.CheckoutLink$outboundSchema = exports.CheckoutLink$inboundSchema = exports.CheckoutLinkDiscount$ = exports.CheckoutLinkDiscount$outboundSchema = exports.CheckoutLinkDiscount$inboundSchema = exports.CheckoutLinkMetadata$ = exports.CheckoutLinkMetadata$outboundSchema = exports.CheckoutLinkMetadata$inboundSchema = void 0;
exports.checkoutLinkMetadataToJSON = checkoutLinkMetadataToJSON;
exports.checkoutLinkMetadataFromJSON = checkoutLinkMetadataFromJSON;
exports.checkoutLinkDiscountToJSON = checkoutLinkDiscountToJSON;
exports.checkoutLinkDiscountFromJSON = checkoutLinkDiscountFromJSON;
exports.checkoutLinkToJSON = checkoutLinkToJSON;
exports.checkoutLinkFromJSON = checkoutLinkFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const checkoutlinkproduct_js_1 = require("./checkoutlinkproduct.js");
const discountfixedonceforeverdurationbase_js_1 = require("./discountfixedonceforeverdurationbase.js");
const discountfixedrepeatdurationbase_js_1 = require("./discountfixedrepeatdurationbase.js");
const discountpercentageonceforeverdurationbase_js_1 = require("./discountpercentageonceforeverdurationbase.js");
const discountpercentagerepeatdurationbase_js_1 = require("./discountpercentagerepeatdurationbase.js");
const productprice_js_1 = require("./productprice.js");
/** @internal */
exports.CheckoutLinkMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
exports.CheckoutLinkMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkMetadata$;
(function (CheckoutLinkMetadata$) {
    /** @deprecated use `CheckoutLinkMetadata$inboundSchema` instead. */
    CheckoutLinkMetadata$.inboundSchema = exports.CheckoutLinkMetadata$inboundSchema;
    /** @deprecated use `CheckoutLinkMetadata$outboundSchema` instead. */
    CheckoutLinkMetadata$.outboundSchema = exports.CheckoutLinkMetadata$outboundSchema;
})(CheckoutLinkMetadata$ || (exports.CheckoutLinkMetadata$ = CheckoutLinkMetadata$ = {}));
function checkoutLinkMetadataToJSON(checkoutLinkMetadata) {
    return JSON.stringify(exports.CheckoutLinkMetadata$outboundSchema.parse(checkoutLinkMetadata));
}
function checkoutLinkMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinkMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkMetadata' from JSON`);
}
/** @internal */
exports.CheckoutLinkDiscount$inboundSchema = z.union([
    discountpercentageonceforeverdurationbase_js_1.DiscountPercentageOnceForeverDurationBase$inboundSchema,
    discountfixedonceforeverdurationbase_js_1.DiscountFixedOnceForeverDurationBase$inboundSchema,
    discountpercentagerepeatdurationbase_js_1.DiscountPercentageRepeatDurationBase$inboundSchema,
    discountfixedrepeatdurationbase_js_1.DiscountFixedRepeatDurationBase$inboundSchema,
]);
/** @internal */
exports.CheckoutLinkDiscount$outboundSchema = z.union([
    discountpercentageonceforeverdurationbase_js_1.DiscountPercentageOnceForeverDurationBase$outboundSchema,
    discountfixedonceforeverdurationbase_js_1.DiscountFixedOnceForeverDurationBase$outboundSchema,
    discountpercentagerepeatdurationbase_js_1.DiscountPercentageRepeatDurationBase$outboundSchema,
    discountfixedrepeatdurationbase_js_1.DiscountFixedRepeatDurationBase$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkDiscount$;
(function (CheckoutLinkDiscount$) {
    /** @deprecated use `CheckoutLinkDiscount$inboundSchema` instead. */
    CheckoutLinkDiscount$.inboundSchema = exports.CheckoutLinkDiscount$inboundSchema;
    /** @deprecated use `CheckoutLinkDiscount$outboundSchema` instead. */
    CheckoutLinkDiscount$.outboundSchema = exports.CheckoutLinkDiscount$outboundSchema;
})(CheckoutLinkDiscount$ || (exports.CheckoutLinkDiscount$ = CheckoutLinkDiscount$ = {}));
function checkoutLinkDiscountToJSON(checkoutLinkDiscount) {
    return JSON.stringify(exports.CheckoutLinkDiscount$outboundSchema.parse(checkoutLinkDiscount));
}
function checkoutLinkDiscountFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinkDiscount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkDiscount' from JSON`);
}
/** @internal */
exports.CheckoutLink$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()])),
    payment_processor: z.literal("stripe").optional(),
    client_secret: z.string(),
    success_url: z.nullable(z.string()),
    label: z.nullable(z.string()),
    allow_discount_codes: z.boolean(),
    product_id: z.string(),
    product_price_id: z.nullable(z.string()),
    discount_id: z.nullable(z.string()),
    product: checkoutlinkproduct_js_1.CheckoutLinkProduct$inboundSchema,
    product_price: z.nullable(productprice_js_1.ProductPrice$inboundSchema),
    discount: z.nullable(z.union([
        discountpercentageonceforeverdurationbase_js_1.DiscountPercentageOnceForeverDurationBase$inboundSchema,
        discountfixedonceforeverdurationbase_js_1.DiscountFixedOnceForeverDurationBase$inboundSchema,
        discountpercentagerepeatdurationbase_js_1.DiscountPercentageRepeatDurationBase$inboundSchema,
        discountfixedrepeatdurationbase_js_1.DiscountFixedRepeatDurationBase$inboundSchema,
    ])),
    url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "payment_processor": "paymentProcessor",
        "client_secret": "clientSecret",
        "success_url": "successUrl",
        "allow_discount_codes": "allowDiscountCodes",
        "product_id": "productId",
        "product_price_id": "productPriceId",
        "discount_id": "discountId",
        "product_price": "productPrice",
    });
});
/** @internal */
exports.CheckoutLink$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()])),
    paymentProcessor: z.literal("stripe").default("stripe"),
    clientSecret: z.string(),
    successUrl: z.nullable(z.string()),
    label: z.nullable(z.string()),
    allowDiscountCodes: z.boolean(),
    productId: z.string(),
    productPriceId: z.nullable(z.string()),
    discountId: z.nullable(z.string()),
    product: checkoutlinkproduct_js_1.CheckoutLinkProduct$outboundSchema,
    productPrice: z.nullable(productprice_js_1.ProductPrice$outboundSchema),
    discount: z.nullable(z.union([
        discountpercentageonceforeverdurationbase_js_1.DiscountPercentageOnceForeverDurationBase$outboundSchema,
        discountfixedonceforeverdurationbase_js_1.DiscountFixedOnceForeverDurationBase$outboundSchema,
        discountpercentagerepeatdurationbase_js_1.DiscountPercentageRepeatDurationBase$outboundSchema,
        discountfixedrepeatdurationbase_js_1.DiscountFixedRepeatDurationBase$outboundSchema,
    ])),
    url: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        paymentProcessor: "payment_processor",
        clientSecret: "client_secret",
        successUrl: "success_url",
        allowDiscountCodes: "allow_discount_codes",
        productId: "product_id",
        productPriceId: "product_price_id",
        discountId: "discount_id",
        productPrice: "product_price",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLink$;
(function (CheckoutLink$) {
    /** @deprecated use `CheckoutLink$inboundSchema` instead. */
    CheckoutLink$.inboundSchema = exports.CheckoutLink$inboundSchema;
    /** @deprecated use `CheckoutLink$outboundSchema` instead. */
    CheckoutLink$.outboundSchema = exports.CheckoutLink$outboundSchema;
})(CheckoutLink$ || (exports.CheckoutLink$ = CheckoutLink$ = {}));
function checkoutLinkToJSON(checkoutLink) {
    return JSON.stringify(exports.CheckoutLink$outboundSchema.parse(checkoutLink));
}
function checkoutLinkFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLink$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLink' from JSON`);
}
