// @polar-sh/sdk@0.18.0/models/components/checkout.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Checkout$ = exports.Checkout$outboundSchema = exports.Checkout$inboundSchema = exports.CheckoutDiscount$ = exports.CheckoutDiscount$outboundSchema = exports.CheckoutDiscount$inboundSchema = exports.CheckoutMetadata$ = exports.CheckoutMetadata$outboundSchema = exports.CheckoutMetadata$inboundSchema = exports.PaymentProcessorMetadata$ = exports.PaymentProcessorMetadata$outboundSchema = exports.PaymentProcessorMetadata$inboundSchema = exports.CheckoutCustomFieldData$ = exports.CheckoutCustomFieldData$outboundSchema = exports.CheckoutCustomFieldData$inboundSchema = void 0;
exports.checkoutCustomFieldDataToJSON = checkoutCustomFieldDataToJSON;
exports.checkoutCustomFieldDataFromJSON = checkoutCustomFieldDataFromJSON;
exports.paymentProcessorMetadataToJSON = paymentProcessorMetadataToJSON;
exports.paymentProcessorMetadataFromJSON = paymentProcessorMetadataFromJSON;
exports.checkoutMetadataToJSON = checkoutMetadataToJSON;
exports.checkoutMetadataFromJSON = checkoutMetadataFromJSON;
exports.checkoutDiscountToJSON = checkoutDiscountToJSON;
exports.checkoutDiscountFromJSON = checkoutDiscountFromJSON;
exports.checkoutToJSON = checkoutToJSON;
exports.checkoutFromJSON = checkoutFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const address_js_1 = require("./address.js");
const attachedcustomfield_js_1 = require("./attachedcustomfield.js");
const checkoutdiscountfixedonceforeverduration_js_1 = require("./checkoutdiscountfixedonceforeverduration.js");
const checkoutdiscountfixedrepeatduration_js_1 = require("./checkoutdiscountfixedrepeatduration.js");
const checkoutdiscountpercentageonceforeverduration_js_1 = require("./checkoutdiscountpercentageonceforeverduration.js");
const checkoutdiscountpercentagerepeatduration_js_1 = require("./checkoutdiscountpercentagerepeatduration.js");
const checkoutproduct_js_1 = require("./checkoutproduct.js");
const checkoutstatus_js_1 = require("./checkoutstatus.js");
const productprice_js_1 = require("./productprice.js");
/** @internal */
exports.CheckoutCustomFieldData$inboundSchema = z.object({});
/** @internal */
exports.CheckoutCustomFieldData$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutCustomFieldData$;
(function (CheckoutCustomFieldData$) {
    /** @deprecated use `CheckoutCustomFieldData$inboundSchema` instead. */
    CheckoutCustomFieldData$.inboundSchema = exports.CheckoutCustomFieldData$inboundSchema;
    /** @deprecated use `CheckoutCustomFieldData$outboundSchema` instead. */
    CheckoutCustomFieldData$.outboundSchema = exports.CheckoutCustomFieldData$outboundSchema;
})(CheckoutCustomFieldData$ || (exports.CheckoutCustomFieldData$ = CheckoutCustomFieldData$ = {}));
function checkoutCustomFieldDataToJSON(checkoutCustomFieldData) {
    return JSON.stringify(exports.CheckoutCustomFieldData$outboundSchema.parse(checkoutCustomFieldData));
}
function checkoutCustomFieldDataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutCustomFieldData' from JSON`);
}
/** @internal */
exports.PaymentProcessorMetadata$inboundSchema = z.object({});
/** @internal */
exports.PaymentProcessorMetadata$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PaymentProcessorMetadata$;
(function (PaymentProcessorMetadata$) {
    /** @deprecated use `PaymentProcessorMetadata$inboundSchema` instead. */
    PaymentProcessorMetadata$.inboundSchema = exports.PaymentProcessorMetadata$inboundSchema;
    /** @deprecated use `PaymentProcessorMetadata$outboundSchema` instead. */
    PaymentProcessorMetadata$.outboundSchema = exports.PaymentProcessorMetadata$outboundSchema;
})(PaymentProcessorMetadata$ || (exports.PaymentProcessorMetadata$ = PaymentProcessorMetadata$ = {}));
function paymentProcessorMetadataToJSON(paymentProcessorMetadata) {
    return JSON.stringify(exports.PaymentProcessorMetadata$outboundSchema.parse(paymentProcessorMetadata));
}
function paymentProcessorMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PaymentProcessorMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PaymentProcessorMetadata' from JSON`);
}
/** @internal */
exports.CheckoutMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
exports.CheckoutMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutMetadata$;
(function (CheckoutMetadata$) {
    /** @deprecated use `CheckoutMetadata$inboundSchema` instead. */
    CheckoutMetadata$.inboundSchema = exports.CheckoutMetadata$inboundSchema;
    /** @deprecated use `CheckoutMetadata$outboundSchema` instead. */
    CheckoutMetadata$.outboundSchema = exports.CheckoutMetadata$outboundSchema;
})(CheckoutMetadata$ || (exports.CheckoutMetadata$ = CheckoutMetadata$ = {}));
function checkoutMetadataToJSON(checkoutMetadata) {
    return JSON.stringify(exports.CheckoutMetadata$outboundSchema.parse(checkoutMetadata));
}
function checkoutMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutMetadata' from JSON`);
}
/** @internal */
exports.CheckoutDiscount$inboundSchema = z.union([
    checkoutdiscountpercentageonceforeverduration_js_1.CheckoutDiscountPercentageOnceForeverDuration$inboundSchema,
    checkoutdiscountfixedonceforeverduration_js_1.CheckoutDiscountFixedOnceForeverDuration$inboundSchema,
    checkoutdiscountpercentagerepeatduration_js_1.CheckoutDiscountPercentageRepeatDuration$inboundSchema,
    checkoutdiscountfixedrepeatduration_js_1.CheckoutDiscountFixedRepeatDuration$inboundSchema,
]);
/** @internal */
exports.CheckoutDiscount$outboundSchema = z.union([
    checkoutdiscountpercentageonceforeverduration_js_1.CheckoutDiscountPercentageOnceForeverDuration$outboundSchema,
    checkoutdiscountfixedonceforeverduration_js_1.CheckoutDiscountFixedOnceForeverDuration$outboundSchema,
    checkoutdiscountpercentagerepeatduration_js_1.CheckoutDiscountPercentageRepeatDuration$outboundSchema,
    checkoutdiscountfixedrepeatduration_js_1.CheckoutDiscountFixedRepeatDuration$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutDiscount$;
(function (CheckoutDiscount$) {
    /** @deprecated use `CheckoutDiscount$inboundSchema` instead. */
    CheckoutDiscount$.inboundSchema = exports.CheckoutDiscount$inboundSchema;
    /** @deprecated use `CheckoutDiscount$outboundSchema` instead. */
    CheckoutDiscount$.outboundSchema = exports.CheckoutDiscount$outboundSchema;
})(CheckoutDiscount$ || (exports.CheckoutDiscount$ = CheckoutDiscount$ = {}));
function checkoutDiscountToJSON(checkoutDiscount) {
    return JSON.stringify(exports.CheckoutDiscount$outboundSchema.parse(checkoutDiscount));
}
function checkoutDiscountFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutDiscount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutDiscount' from JSON`);
}
/** @internal */
exports.Checkout$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    custom_field_data: z.lazy(() => exports.CheckoutCustomFieldData$inboundSchema)
        .optional(),
    payment_processor: z.literal("stripe").optional(),
    status: checkoutstatus_js_1.CheckoutStatus$inboundSchema,
    client_secret: z.string(),
    url: z.string(),
    expires_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    success_url: z.string(),
    embed_origin: z.nullable(z.string()),
    amount: z.nullable(z.number().int()),
    tax_amount: z.nullable(z.number().int()),
    currency: z.nullable(z.string()),
    subtotal_amount: z.nullable(z.number().int()),
    total_amount: z.nullable(z.number().int()),
    product_id: z.string(),
    product_price_id: z.string(),
    discount_id: z.nullable(z.string()),
    allow_discount_codes: z.boolean(),
    is_discount_applicable: z.boolean(),
    is_free_product_price: z.boolean(),
    is_payment_required: z.boolean(),
    is_payment_setup_required: z.boolean(),
    is_payment_form_required: z.boolean(),
    customer_id: z.nullable(z.string()),
    customer_name: z.nullable(z.string()),
    customer_email: z.nullable(z.string()),
    customer_ip_address: z.nullable(z.string()),
    customer_billing_address: z.nullable(address_js_1.Address$inboundSchema),
    customer_tax_id: z.nullable(z.string()),
    payment_processor_metadata: z.lazy(() => exports.PaymentProcessorMetadata$inboundSchema),
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()])),
    product: checkoutproduct_js_1.CheckoutProduct$inboundSchema,
    product_price: productprice_js_1.ProductPrice$inboundSchema,
    discount: z.nullable(z.union([
        checkoutdiscountpercentageonceforeverduration_js_1.CheckoutDiscountPercentageOnceForeverDuration$inboundSchema,
        checkoutdiscountfixedonceforeverduration_js_1.CheckoutDiscountFixedOnceForeverDuration$inboundSchema,
        checkoutdiscountpercentagerepeatduration_js_1.CheckoutDiscountPercentageRepeatDuration$inboundSchema,
        checkoutdiscountfixedrepeatduration_js_1.CheckoutDiscountFixedRepeatDuration$inboundSchema,
    ])),
    subscription_id: z.nullable(z.string()),
    attached_custom_fields: z.array(attachedcustomfield_js_1.AttachedCustomField$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "custom_field_data": "customFieldData",
        "payment_processor": "paymentProcessor",
        "client_secret": "clientSecret",
        "expires_at": "expiresAt",
        "success_url": "successUrl",
        "embed_origin": "embedOrigin",
        "tax_amount": "taxAmount",
        "subtotal_amount": "subtotalAmount",
        "total_amount": "totalAmount",
        "product_id": "productId",
        "product_price_id": "productPriceId",
        "discount_id": "discountId",
        "allow_discount_codes": "allowDiscountCodes",
        "is_discount_applicable": "isDiscountApplicable",
        "is_free_product_price": "isFreeProductPrice",
        "is_payment_required": "isPaymentRequired",
        "is_payment_setup_required": "isPaymentSetupRequired",
        "is_payment_form_required": "isPaymentFormRequired",
        "customer_id": "customerId",
        "customer_name": "customerName",
        "customer_email": "customerEmail",
        "customer_ip_address": "customerIpAddress",
        "customer_billing_address": "customerBillingAddress",
        "customer_tax_id": "customerTaxId",
        "payment_processor_metadata": "paymentProcessorMetadata",
        "product_price": "productPrice",
        "subscription_id": "subscriptionId",
        "attached_custom_fields": "attachedCustomFields",
    });
});
/** @internal */
exports.Checkout$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    customFieldData: z.lazy(() => exports.CheckoutCustomFieldData$outboundSchema)
        .optional(),
    paymentProcessor: z.literal("stripe").default("stripe"),
    status: checkoutstatus_js_1.CheckoutStatus$outboundSchema,
    clientSecret: z.string(),
    url: z.string(),
    expiresAt: z.date().transform(v => v.toISOString()),
    successUrl: z.string(),
    embedOrigin: z.nullable(z.string()),
    amount: z.nullable(z.number().int()),
    taxAmount: z.nullable(z.number().int()),
    currency: z.nullable(z.string()),
    subtotalAmount: z.nullable(z.number().int()),
    totalAmount: z.nullable(z.number().int()),
    productId: z.string(),
    productPriceId: z.string(),
    discountId: z.nullable(z.string()),
    allowDiscountCodes: z.boolean(),
    isDiscountApplicable: z.boolean(),
    isFreeProductPrice: z.boolean(),
    isPaymentRequired: z.boolean(),
    isPaymentSetupRequired: z.boolean(),
    isPaymentFormRequired: z.boolean(),
    customerId: z.nullable(z.string()),
    customerName: z.nullable(z.string()),
    customerEmail: z.nullable(z.string()),
    customerIpAddress: z.nullable(z.string()),
    customerBillingAddress: z.nullable(address_js_1.Address$outboundSchema),
    customerTaxId: z.nullable(z.string()),
    paymentProcessorMetadata: z.lazy(() => exports.PaymentProcessorMetadata$outboundSchema),
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()])),
    product: checkoutproduct_js_1.CheckoutProduct$outboundSchema,
    productPrice: productprice_js_1.ProductPrice$outboundSchema,
    discount: z.nullable(z.union([
        checkoutdiscountpercentageonceforeverduration_js_1.CheckoutDiscountPercentageOnceForeverDuration$outboundSchema,
        checkoutdiscountfixedonceforeverduration_js_1.CheckoutDiscountFixedOnceForeverDuration$outboundSchema,
        checkoutdiscountpercentagerepeatduration_js_1.CheckoutDiscountPercentageRepeatDuration$outboundSchema,
        checkoutdiscountfixedrepeatduration_js_1.CheckoutDiscountFixedRepeatDuration$outboundSchema,
    ])),
    subscriptionId: z.nullable(z.string()),
    attachedCustomFields: z.array(attachedcustomfield_js_1.AttachedCustomField$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        customFieldData: "custom_field_data",
        paymentProcessor: "payment_processor",
        clientSecret: "client_secret",
        expiresAt: "expires_at",
        successUrl: "success_url",
        embedOrigin: "embed_origin",
        taxAmount: "tax_amount",
        subtotalAmount: "subtotal_amount",
        totalAmount: "total_amount",
        productId: "product_id",
        productPriceId: "product_price_id",
        discountId: "discount_id",
        allowDiscountCodes: "allow_discount_codes",
        isDiscountApplicable: "is_discount_applicable",
        isFreeProductPrice: "is_free_product_price",
        isPaymentRequired: "is_payment_required",
        isPaymentSetupRequired: "is_payment_setup_required",
        isPaymentFormRequired: "is_payment_form_required",
        customerId: "customer_id",
        customerName: "customer_name",
        customerEmail: "customer_email",
        customerIpAddress: "customer_ip_address",
        customerBillingAddress: "customer_billing_address",
        customerTaxId: "customer_tax_id",
        paymentProcessorMetadata: "payment_processor_metadata",
        productPrice: "product_price",
        subscriptionId: "subscription_id",
        attachedCustomFields: "attached_custom_fields",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Checkout$;
(function (Checkout$) {
    /** @deprecated use `Checkout$inboundSchema` instead. */
    Checkout$.inboundSchema = exports.Checkout$inboundSchema;
    /** @deprecated use `Checkout$outboundSchema` instead. */
    Checkout$.outboundSchema = exports.Checkout$outboundSchema;
})(Checkout$ || (exports.Checkout$ = Checkout$ = {}));
function checkoutToJSON(checkout) {
    return JSON.stringify(exports.Checkout$outboundSchema.parse(checkout));
}
function checkoutFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Checkout$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Checkout' from JSON`);
}
