// @polar-sh/sdk@0.18.0/models/components/checkoutlinkproductcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutLinkProductCreate$ = exports.CheckoutLinkProductCreate$outboundSchema = exports.CheckoutLinkProductCreate$inboundSchema = exports.CheckoutLinkProductCreatePaymentProcessor$ = exports.CheckoutLinkProductCreatePaymentProcessor$outboundSchema = exports.CheckoutLinkProductCreatePaymentProcessor$inboundSchema = exports.CheckoutLinkProductCreateMetadata$ = exports.CheckoutLinkProductCreateMetadata$outboundSchema = exports.CheckoutLinkProductCreateMetadata$inboundSchema = exports.CheckoutLinkProductCreatePaymentProcessor = void 0;
exports.checkoutLinkProductCreateMetadataToJSON = checkoutLinkProductCreateMetadataToJSON;
exports.checkoutLinkProductCreateMetadataFromJSON = checkoutLinkProductCreateMetadataFromJSON;
exports.checkoutLinkProductCreateToJSON = checkoutLinkProductCreateToJSON;
exports.checkoutLinkProductCreateFromJSON = checkoutLinkProductCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * Payment processor to use. Currently only Stripe is supported.
 */
exports.CheckoutLinkProductCreatePaymentProcessor = {
    Stripe: "stripe",
};
/** @internal */
exports.CheckoutLinkProductCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
exports.CheckoutLinkProductCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkProductCreateMetadata$;
(function (CheckoutLinkProductCreateMetadata$) {
    /** @deprecated use `CheckoutLinkProductCreateMetadata$inboundSchema` instead. */
    CheckoutLinkProductCreateMetadata$.inboundSchema = exports.CheckoutLinkProductCreateMetadata$inboundSchema;
    /** @deprecated use `CheckoutLinkProductCreateMetadata$outboundSchema` instead. */
    CheckoutLinkProductCreateMetadata$.outboundSchema = exports.CheckoutLinkProductCreateMetadata$outboundSchema;
})(CheckoutLinkProductCreateMetadata$ || (exports.CheckoutLinkProductCreateMetadata$ = CheckoutLinkProductCreateMetadata$ = {}));
function checkoutLinkProductCreateMetadataToJSON(checkoutLinkProductCreateMetadata) {
    return JSON.stringify(exports.CheckoutLinkProductCreateMetadata$outboundSchema.parse(checkoutLinkProductCreateMetadata));
}
function checkoutLinkProductCreateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinkProductCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkProductCreateMetadata' from JSON`);
}
/** @internal */
exports.CheckoutLinkProductCreatePaymentProcessor$inboundSchema = z
    .nativeEnum(exports.CheckoutLinkProductCreatePaymentProcessor);
/** @internal */
exports.CheckoutLinkProductCreatePaymentProcessor$outboundSchema = exports.CheckoutLinkProductCreatePaymentProcessor$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkProductCreatePaymentProcessor$;
(function (CheckoutLinkProductCreatePaymentProcessor$) {
    /** @deprecated use `CheckoutLinkProductCreatePaymentProcessor$inboundSchema` instead. */
    CheckoutLinkProductCreatePaymentProcessor$.inboundSchema = exports.CheckoutLinkProductCreatePaymentProcessor$inboundSchema;
    /** @deprecated use `CheckoutLinkProductCreatePaymentProcessor$outboundSchema` instead. */
    CheckoutLinkProductCreatePaymentProcessor$.outboundSchema = exports.CheckoutLinkProductCreatePaymentProcessor$outboundSchema;
})(CheckoutLinkProductCreatePaymentProcessor$ || (exports.CheckoutLinkProductCreatePaymentProcessor$ = CheckoutLinkProductCreatePaymentProcessor$ = {}));
/** @internal */
exports.CheckoutLinkProductCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    payment_processor: z.literal("stripe").optional(),
    label: z.nullable(z.string()).optional(),
    allow_discount_codes: z.boolean().default(true),
    discount_id: z.nullable(z.string()).optional(),
    success_url: z.nullable(z.string()).optional(),
    product_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "payment_processor": "paymentProcessor",
        "allow_discount_codes": "allowDiscountCodes",
        "discount_id": "discountId",
        "success_url": "successUrl",
        "product_id": "productId",
    });
});
/** @internal */
exports.CheckoutLinkProductCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    paymentProcessor: z.literal("stripe").default("stripe"),
    label: z.nullable(z.string()).optional(),
    allowDiscountCodes: z.boolean().default(true),
    discountId: z.nullable(z.string()).optional(),
    successUrl: z.nullable(z.string()).optional(),
    productId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        paymentProcessor: "payment_processor",
        allowDiscountCodes: "allow_discount_codes",
        discountId: "discount_id",
        successUrl: "success_url",
        productId: "product_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkProductCreate$;
(function (CheckoutLinkProductCreate$) {
    /** @deprecated use `CheckoutLinkProductCreate$inboundSchema` instead. */
    CheckoutLinkProductCreate$.inboundSchema = exports.CheckoutLinkProductCreate$inboundSchema;
    /** @deprecated use `CheckoutLinkProductCreate$outboundSchema` instead. */
    CheckoutLinkProductCreate$.outboundSchema = exports.CheckoutLinkProductCreate$outboundSchema;
})(CheckoutLinkProductCreate$ || (exports.CheckoutLinkProductCreate$ = CheckoutLinkProductCreate$ = {}));
function checkoutLinkProductCreateToJSON(checkoutLinkProductCreate) {
    return JSON.stringify(exports.CheckoutLinkProductCreate$outboundSchema.parse(checkoutLinkProductCreate));
}
function checkoutLinkProductCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinkProductCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkProductCreate' from JSON`);
}
