// @polar-sh/sdk@0.18.0/models/components/checkoutlinkproduct.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutLinkProduct$ = exports.CheckoutLinkProduct$outboundSchema = exports.CheckoutLinkProduct$inboundSchema = void 0;
exports.checkoutLinkProductToJSON = checkoutLinkProductToJSON;
exports.checkoutLinkProductFromJSON = checkoutLinkProductFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const benefitbase_js_1 = require("./benefitbase.js");
const productmediafileread_js_1 = require("./productmediafileread.js");
const productprice_js_1 = require("./productprice.js");
/** @internal */
exports.CheckoutLinkProduct$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    is_recurring: z.boolean(),
    is_archived: z.boolean(),
    organization_id: z.string(),
    prices: z.array(productprice_js_1.ProductPrice$inboundSchema),
    benefits: z.array(benefitbase_js_1.BenefitBase$inboundSchema),
    medias: z.array(productmediafileread_js_1.ProductMediaFileRead$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "is_recurring": "isRecurring",
        "is_archived": "isArchived",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.CheckoutLinkProduct$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    isRecurring: z.boolean(),
    isArchived: z.boolean(),
    organizationId: z.string(),
    prices: z.array(productprice_js_1.ProductPrice$outboundSchema),
    benefits: z.array(benefitbase_js_1.BenefitBase$outboundSchema),
    medias: z.array(productmediafileread_js_1.ProductMediaFileRead$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        isRecurring: "is_recurring",
        isArchived: "is_archived",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutLinkProduct$;
(function (CheckoutLinkProduct$) {
    /** @deprecated use `CheckoutLinkProduct$inboundSchema` instead. */
    CheckoutLinkProduct$.inboundSchema = exports.CheckoutLinkProduct$inboundSchema;
    /** @deprecated use `CheckoutLinkProduct$outboundSchema` instead. */
    CheckoutLinkProduct$.outboundSchema = exports.CheckoutLinkProduct$outboundSchema;
})(CheckoutLinkProduct$ || (exports.CheckoutLinkProduct$ = CheckoutLinkProduct$ = {}));
function checkoutLinkProductToJSON(checkoutLinkProduct) {
    return JSON.stringify(exports.CheckoutLinkProduct$outboundSchema.parse(checkoutLinkProduct));
}
function checkoutLinkProductFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutLinkProduct$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutLinkProduct' from JSON`);
}
