// @polar-sh/sdk@0.18.0/models/components/checkoutproductcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutProductCreate$ = exports.CheckoutProductCreate$outboundSchema = exports.CheckoutProductCreate$inboundSchema = exports.CheckoutProductCreatePaymentProcessor$ = exports.CheckoutProductCreatePaymentProcessor$outboundSchema = exports.CheckoutProductCreatePaymentProcessor$inboundSchema = exports.CheckoutProductCreateCustomFieldData$ = exports.CheckoutProductCreateCustomFieldData$outboundSchema = exports.CheckoutProductCreateCustomFieldData$inboundSchema = exports.CheckoutProductCreateMetadata$ = exports.CheckoutProductCreateMetadata$outboundSchema = exports.CheckoutProductCreateMetadata$inboundSchema = exports.CheckoutProductCreatePaymentProcessor = void 0;
exports.checkoutProductCreateMetadataToJSON = checkoutProductCreateMetadataToJSON;
exports.checkoutProductCreateMetadataFromJSON = checkoutProductCreateMetadataFromJSON;
exports.checkoutProductCreateCustomFieldDataToJSON = checkoutProductCreateCustomFieldDataToJSON;
exports.checkoutProductCreateCustomFieldDataFromJSON = checkoutProductCreateCustomFieldDataFromJSON;
exports.checkoutProductCreateToJSON = checkoutProductCreateToJSON;
exports.checkoutProductCreateFromJSON = checkoutProductCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const address_js_1 = require("./address.js");
/**
 * Payment processor to use. Currently only Stripe is supported.
 */
exports.CheckoutProductCreatePaymentProcessor = {
    Stripe: "stripe",
};
/** @internal */
exports.CheckoutProductCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
exports.CheckoutProductCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutProductCreateMetadata$;
(function (CheckoutProductCreateMetadata$) {
    /** @deprecated use `CheckoutProductCreateMetadata$inboundSchema` instead. */
    CheckoutProductCreateMetadata$.inboundSchema = exports.CheckoutProductCreateMetadata$inboundSchema;
    /** @deprecated use `CheckoutProductCreateMetadata$outboundSchema` instead. */
    CheckoutProductCreateMetadata$.outboundSchema = exports.CheckoutProductCreateMetadata$outboundSchema;
})(CheckoutProductCreateMetadata$ || (exports.CheckoutProductCreateMetadata$ = CheckoutProductCreateMetadata$ = {}));
function checkoutProductCreateMetadataToJSON(checkoutProductCreateMetadata) {
    return JSON.stringify(exports.CheckoutProductCreateMetadata$outboundSchema.parse(checkoutProductCreateMetadata));
}
function checkoutProductCreateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutProductCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutProductCreateMetadata' from JSON`);
}
/** @internal */
exports.CheckoutProductCreateCustomFieldData$inboundSchema = z.object({});
/** @internal */
exports.CheckoutProductCreateCustomFieldData$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutProductCreateCustomFieldData$;
(function (CheckoutProductCreateCustomFieldData$) {
    /** @deprecated use `CheckoutProductCreateCustomFieldData$inboundSchema` instead. */
    CheckoutProductCreateCustomFieldData$.inboundSchema = exports.CheckoutProductCreateCustomFieldData$inboundSchema;
    /** @deprecated use `CheckoutProductCreateCustomFieldData$outboundSchema` instead. */
    CheckoutProductCreateCustomFieldData$.outboundSchema = exports.CheckoutProductCreateCustomFieldData$outboundSchema;
})(CheckoutProductCreateCustomFieldData$ || (exports.CheckoutProductCreateCustomFieldData$ = CheckoutProductCreateCustomFieldData$ = {}));
function checkoutProductCreateCustomFieldDataToJSON(checkoutProductCreateCustomFieldData) {
    return JSON.stringify(exports.CheckoutProductCreateCustomFieldData$outboundSchema.parse(checkoutProductCreateCustomFieldData));
}
function checkoutProductCreateCustomFieldDataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutProductCreateCustomFieldData$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutProductCreateCustomFieldData' from JSON`);
}
/** @internal */
exports.CheckoutProductCreatePaymentProcessor$inboundSchema = z.nativeEnum(exports.CheckoutProductCreatePaymentProcessor);
/** @internal */
exports.CheckoutProductCreatePaymentProcessor$outboundSchema = exports.CheckoutProductCreatePaymentProcessor$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutProductCreatePaymentProcessor$;
(function (CheckoutProductCreatePaymentProcessor$) {
    /** @deprecated use `CheckoutProductCreatePaymentProcessor$inboundSchema` instead. */
    CheckoutProductCreatePaymentProcessor$.inboundSchema = exports.CheckoutProductCreatePaymentProcessor$inboundSchema;
    /** @deprecated use `CheckoutProductCreatePaymentProcessor$outboundSchema` instead. */
    CheckoutProductCreatePaymentProcessor$.outboundSchema = exports.CheckoutProductCreatePaymentProcessor$outboundSchema;
})(CheckoutProductCreatePaymentProcessor$ || (exports.CheckoutProductCreatePaymentProcessor$ = CheckoutProductCreatePaymentProcessor$ = {}));
/** @internal */
exports.CheckoutProductCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    custom_field_data: z.lazy(() => exports.CheckoutProductCreateCustomFieldData$inboundSchema).optional(),
    payment_processor: z.literal("stripe").optional(),
    discount_id: z.nullable(z.string()).optional(),
    allow_discount_codes: z.boolean().default(true),
    amount: z.nullable(z.number().int()).optional(),
    customer_name: z.nullable(z.string()).optional(),
    customer_email: z.nullable(z.string()).optional(),
    customer_ip_address: z.nullable(z.string()).optional(),
    customer_billing_address: z.nullable(address_js_1.Address$inboundSchema).optional(),
    customer_tax_id: z.nullable(z.string()).optional(),
    subscription_id: z.nullable(z.string()).optional(),
    success_url: z.nullable(z.string()).optional(),
    embed_origin: z.nullable(z.string()).optional(),
    product_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "custom_field_data": "customFieldData",
        "payment_processor": "paymentProcessor",
        "discount_id": "discountId",
        "allow_discount_codes": "allowDiscountCodes",
        "customer_name": "customerName",
        "customer_email": "customerEmail",
        "customer_ip_address": "customerIpAddress",
        "customer_billing_address": "customerBillingAddress",
        "customer_tax_id": "customerTaxId",
        "subscription_id": "subscriptionId",
        "success_url": "successUrl",
        "embed_origin": "embedOrigin",
        "product_id": "productId",
    });
});
/** @internal */
exports.CheckoutProductCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    customFieldData: z.lazy(() => exports.CheckoutProductCreateCustomFieldData$outboundSchema).optional(),
    paymentProcessor: z.literal("stripe").default("stripe"),
    discountId: z.nullable(z.string()).optional(),
    allowDiscountCodes: z.boolean().default(true),
    amount: z.nullable(z.number().int()).optional(),
    customerName: z.nullable(z.string()).optional(),
    customerEmail: z.nullable(z.string()).optional(),
    customerIpAddress: z.nullable(z.string()).optional(),
    customerBillingAddress: z.nullable(address_js_1.Address$outboundSchema).optional(),
    customerTaxId: z.nullable(z.string()).optional(),
    subscriptionId: z.nullable(z.string()).optional(),
    successUrl: z.nullable(z.string()).optional(),
    embedOrigin: z.nullable(z.string()).optional(),
    productId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customFieldData: "custom_field_data",
        paymentProcessor: "payment_processor",
        discountId: "discount_id",
        allowDiscountCodes: "allow_discount_codes",
        customerName: "customer_name",
        customerEmail: "customer_email",
        customerIpAddress: "customer_ip_address",
        customerBillingAddress: "customer_billing_address",
        customerTaxId: "customer_tax_id",
        subscriptionId: "subscription_id",
        successUrl: "success_url",
        embedOrigin: "embed_origin",
        productId: "product_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CheckoutProductCreate$;
(function (CheckoutProductCreate$) {
    /** @deprecated use `CheckoutProductCreate$inboundSchema` instead. */
    CheckoutProductCreate$.inboundSchema = exports.CheckoutProductCreate$inboundSchema;
    /** @deprecated use `CheckoutProductCreate$outboundSchema` instead. */
    CheckoutProductCreate$.outboundSchema = exports.CheckoutProductCreate$outboundSchema;
})(CheckoutProductCreate$ || (exports.CheckoutProductCreate$ = CheckoutProductCreate$ = {}));
function checkoutProductCreateToJSON(checkoutProductCreate) {
    return JSON.stringify(exports.CheckoutProductCreate$outboundSchema.parse(checkoutProductCreate));
}
function checkoutProductCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CheckoutProductCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CheckoutProductCreate' from JSON`);
}
