// @polar-sh/sdk@0.18.0/models/components/customfieldnumberproperties.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldNumberProperties$ = exports.CustomFieldNumberProperties$outboundSchema = exports.CustomFieldNumberProperties$inboundSchema = void 0;
exports.customFieldNumberPropertiesToJSON = customFieldNumberPropertiesToJSON;
exports.customFieldNumberPropertiesFromJSON = customFieldNumberPropertiesFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomFieldNumberProperties$inboundSchema = z.object({
    form_label: z.string().optional(),
    form_help_text: z.string().optional(),
    form_placeholder: z.string().optional(),
    ge: z.number().int().optional(),
    le: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "form_label": "formLabel",
        "form_help_text": "formHelpText",
        "form_placeholder": "formPlaceholder",
    });
});
/** @internal */
exports.CustomFieldNumberProperties$outboundSchema = z.object({
    formLabel: z.string().optional(),
    formHelpText: z.string().optional(),
    formPlaceholder: z.string().optional(),
    ge: z.number().int().optional(),
    le: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        formLabel: "form_label",
        formHelpText: "form_help_text",
        formPlaceholder: "form_placeholder",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldNumberProperties$;
(function (CustomFieldNumberProperties$) {
    /** @deprecated use `CustomFieldNumberProperties$inboundSchema` instead. */
    CustomFieldNumberProperties$.inboundSchema = exports.CustomFieldNumberProperties$inboundSchema;
    /** @deprecated use `CustomFieldNumberProperties$outboundSchema` instead. */
    CustomFieldNumberProperties$.outboundSchema = exports.CustomFieldNumberProperties$outboundSchema;
})(CustomFieldNumberProperties$ || (exports.CustomFieldNumberProperties$ = CustomFieldNumberProperties$ = {}));
function customFieldNumberPropertiesToJSON(customFieldNumberProperties) {
    return JSON.stringify(exports.CustomFieldNumberProperties$outboundSchema.parse(customFieldNumberProperties));
}
function customFieldNumberPropertiesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldNumberProperties$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldNumberProperties' from JSON`);
}
