// @polar-sh/sdk@0.18.0/models/components/customfieldupdatecheckbox.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldUpdateCheckbox$ = exports.CustomFieldUpdateCheckbox$outboundSchema = exports.CustomFieldUpdateCheckbox$inboundSchema = exports.CustomFieldUpdateCheckboxType$ = exports.CustomFieldUpdateCheckboxType$outboundSchema = exports.CustomFieldUpdateCheckboxType$inboundSchema = exports.CustomFieldUpdateCheckboxMetadata$ = exports.CustomFieldUpdateCheckboxMetadata$outboundSchema = exports.CustomFieldUpdateCheckboxMetadata$inboundSchema = exports.CustomFieldUpdateCheckboxType = void 0;
exports.customFieldUpdateCheckboxMetadataToJSON = customFieldUpdateCheckboxMetadataToJSON;
exports.customFieldUpdateCheckboxMetadataFromJSON = customFieldUpdateCheckboxMetadataFromJSON;
exports.customFieldUpdateCheckboxToJSON = customFieldUpdateCheckboxToJSON;
exports.customFieldUpdateCheckboxFromJSON = customFieldUpdateCheckboxFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const customfieldcheckboxproperties_js_1 = require("./customfieldcheckboxproperties.js");
exports.CustomFieldUpdateCheckboxType = {
    Checkbox: "checkbox",
};
/** @internal */
exports.CustomFieldUpdateCheckboxMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
exports.CustomFieldUpdateCheckboxMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldUpdateCheckboxMetadata$;
(function (CustomFieldUpdateCheckboxMetadata$) {
    /** @deprecated use `CustomFieldUpdateCheckboxMetadata$inboundSchema` instead. */
    CustomFieldUpdateCheckboxMetadata$.inboundSchema = exports.CustomFieldUpdateCheckboxMetadata$inboundSchema;
    /** @deprecated use `CustomFieldUpdateCheckboxMetadata$outboundSchema` instead. */
    CustomFieldUpdateCheckboxMetadata$.outboundSchema = exports.CustomFieldUpdateCheckboxMetadata$outboundSchema;
})(CustomFieldUpdateCheckboxMetadata$ || (exports.CustomFieldUpdateCheckboxMetadata$ = CustomFieldUpdateCheckboxMetadata$ = {}));
function customFieldUpdateCheckboxMetadataToJSON(customFieldUpdateCheckboxMetadata) {
    return JSON.stringify(exports.CustomFieldUpdateCheckboxMetadata$outboundSchema.parse(customFieldUpdateCheckboxMetadata));
}
function customFieldUpdateCheckboxMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldUpdateCheckboxMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldUpdateCheckboxMetadata' from JSON`);
}
/** @internal */
exports.CustomFieldUpdateCheckboxType$inboundSchema = z.nativeEnum(exports.CustomFieldUpdateCheckboxType);
/** @internal */
exports.CustomFieldUpdateCheckboxType$outboundSchema = exports.CustomFieldUpdateCheckboxType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldUpdateCheckboxType$;
(function (CustomFieldUpdateCheckboxType$) {
    /** @deprecated use `CustomFieldUpdateCheckboxType$inboundSchema` instead. */
    CustomFieldUpdateCheckboxType$.inboundSchema = exports.CustomFieldUpdateCheckboxType$inboundSchema;
    /** @deprecated use `CustomFieldUpdateCheckboxType$outboundSchema` instead. */
    CustomFieldUpdateCheckboxType$.outboundSchema = exports.CustomFieldUpdateCheckboxType$outboundSchema;
})(CustomFieldUpdateCheckboxType$ || (exports.CustomFieldUpdateCheckboxType$ = CustomFieldUpdateCheckboxType$ = {}));
/** @internal */
exports.CustomFieldUpdateCheckbox$inboundSchema = z.object({
    metadata: z.nullable(z.record(z.union([z.string(), z.number().int(), z.boolean()]))).optional(),
    name: z.nullable(z.string()).optional(),
    slug: z.nullable(z.string()).optional(),
    type: z.literal("checkbox").optional(),
    properties: z.nullable(customfieldcheckboxproperties_js_1.CustomFieldCheckboxProperties$inboundSchema)
        .optional(),
});
/** @internal */
exports.CustomFieldUpdateCheckbox$outboundSchema = z.object({
    metadata: z.nullable(z.record(z.union([z.string(), z.number().int(), z.boolean()]))).optional(),
    name: z.nullable(z.string()).optional(),
    slug: z.nullable(z.string()).optional(),
    type: z.literal("checkbox").default("checkbox"),
    properties: z.nullable(customfieldcheckboxproperties_js_1.CustomFieldCheckboxProperties$outboundSchema)
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldUpdateCheckbox$;
(function (CustomFieldUpdateCheckbox$) {
    /** @deprecated use `CustomFieldUpdateCheckbox$inboundSchema` instead. */
    CustomFieldUpdateCheckbox$.inboundSchema = exports.CustomFieldUpdateCheckbox$inboundSchema;
    /** @deprecated use `CustomFieldUpdateCheckbox$outboundSchema` instead. */
    CustomFieldUpdateCheckbox$.outboundSchema = exports.CustomFieldUpdateCheckbox$outboundSchema;
})(CustomFieldUpdateCheckbox$ || (exports.CustomFieldUpdateCheckbox$ = CustomFieldUpdateCheckbox$ = {}));
function customFieldUpdateCheckboxToJSON(customFieldUpdateCheckbox) {
    return JSON.stringify(exports.CustomFieldUpdateCheckbox$outboundSchema.parse(customFieldUpdateCheckbox));
}
function customFieldUpdateCheckboxFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldUpdateCheckbox$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldUpdateCheckbox' from JSON`);
}
