// @polar-sh/sdk@0.18.0/models/components/discountsortproperty.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscountSortProperty$ = exports.DiscountSortProperty$outboundSchema = exports.DiscountSortProperty$inboundSchema = exports.DiscountSortProperty = void 0;
const z = __importStar(require("zod"));
exports.DiscountSortProperty = {
    CreatedAt: "created_at",
    MinusCreatedAt: "-created_at",
    Name: "name",
    MinusName: "-name",
    Code: "code",
    MinusCode: "-code",
    RedemptionsCount: "redemptions_count",
    MinusRedemptionsCount: "-redemptions_count",
};
/** @internal */
exports.DiscountSortProperty$inboundSchema = z.nativeEnum(exports.DiscountSortProperty);
/** @internal */
exports.DiscountSortProperty$outboundSchema = exports.DiscountSortProperty$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountSortProperty$;
(function (DiscountSortProperty$) {
    /** @deprecated use `DiscountSortProperty$inboundSchema` instead. */
    DiscountSortProperty$.inboundSchema = exports.DiscountSortProperty$inboundSchema;
    /** @deprecated use `DiscountSortProperty$outboundSchema` instead. */
    DiscountSortProperty$.outboundSchema = exports.DiscountSortProperty$outboundSchema;
})(DiscountSortProperty$ || (exports.DiscountSortProperty$ = DiscountSortProperty$ = {}));
