// @polar-sh/sdk@0.18.0/models/components/licensekeyactivate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseKeyActivate$ = exports.LicenseKeyActivate$outboundSchema = exports.LicenseKeyActivate$inboundSchema = exports.LicenseKeyActivateMeta$ = exports.LicenseKeyActivateMeta$outboundSchema = exports.LicenseKeyActivateMeta$inboundSchema = exports.LicenseKeyActivateConditions$ = exports.LicenseKeyActivateConditions$outboundSchema = exports.LicenseKeyActivateConditions$inboundSchema = void 0;
exports.licenseKeyActivateConditionsToJSON = licenseKeyActivateConditionsToJSON;
exports.licenseKeyActivateConditionsFromJSON = licenseKeyActivateConditionsFromJSON;
exports.licenseKeyActivateMetaToJSON = licenseKeyActivateMetaToJSON;
exports.licenseKeyActivateMetaFromJSON = licenseKeyActivateMetaFromJSON;
exports.licenseKeyActivateToJSON = licenseKeyActivateToJSON;
exports.licenseKeyActivateFromJSON = licenseKeyActivateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.LicenseKeyActivateConditions$inboundSchema = z.object({});
/** @internal */
exports.LicenseKeyActivateConditions$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LicenseKeyActivateConditions$;
(function (LicenseKeyActivateConditions$) {
    /** @deprecated use `LicenseKeyActivateConditions$inboundSchema` instead. */
    LicenseKeyActivateConditions$.inboundSchema = exports.LicenseKeyActivateConditions$inboundSchema;
    /** @deprecated use `LicenseKeyActivateConditions$outboundSchema` instead. */
    LicenseKeyActivateConditions$.outboundSchema = exports.LicenseKeyActivateConditions$outboundSchema;
})(LicenseKeyActivateConditions$ || (exports.LicenseKeyActivateConditions$ = LicenseKeyActivateConditions$ = {}));
function licenseKeyActivateConditionsToJSON(licenseKeyActivateConditions) {
    return JSON.stringify(exports.LicenseKeyActivateConditions$outboundSchema.parse(licenseKeyActivateConditions));
}
function licenseKeyActivateConditionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LicenseKeyActivateConditions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LicenseKeyActivateConditions' from JSON`);
}
/** @internal */
exports.LicenseKeyActivateMeta$inboundSchema = z.object({});
/** @internal */
exports.LicenseKeyActivateMeta$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LicenseKeyActivateMeta$;
(function (LicenseKeyActivateMeta$) {
    /** @deprecated use `LicenseKeyActivateMeta$inboundSchema` instead. */
    LicenseKeyActivateMeta$.inboundSchema = exports.LicenseKeyActivateMeta$inboundSchema;
    /** @deprecated use `LicenseKeyActivateMeta$outboundSchema` instead. */
    LicenseKeyActivateMeta$.outboundSchema = exports.LicenseKeyActivateMeta$outboundSchema;
})(LicenseKeyActivateMeta$ || (exports.LicenseKeyActivateMeta$ = LicenseKeyActivateMeta$ = {}));
function licenseKeyActivateMetaToJSON(licenseKeyActivateMeta) {
    return JSON.stringify(exports.LicenseKeyActivateMeta$outboundSchema.parse(licenseKeyActivateMeta));
}
function licenseKeyActivateMetaFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LicenseKeyActivateMeta$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LicenseKeyActivateMeta' from JSON`);
}
/** @internal */
exports.LicenseKeyActivate$inboundSchema = z.object({
    key: z.string(),
    organization_id: z.string(),
    label: z.string(),
    conditions: z.lazy(() => exports.LicenseKeyActivateConditions$inboundSchema)
        .optional(),
    meta: z.lazy(() => exports.LicenseKeyActivateMeta$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.LicenseKeyActivate$outboundSchema = z.object({
    key: z.string(),
    organizationId: z.string(),
    label: z.string(),
    conditions: z.lazy(() => exports.LicenseKeyActivateConditions$outboundSchema)
        .optional(),
    meta: z.lazy(() => exports.LicenseKeyActivateMeta$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LicenseKeyActivate$;
(function (LicenseKeyActivate$) {
    /** @deprecated use `LicenseKeyActivate$inboundSchema` instead. */
    LicenseKeyActivate$.inboundSchema = exports.LicenseKeyActivate$inboundSchema;
    /** @deprecated use `LicenseKeyActivate$outboundSchema` instead. */
    LicenseKeyActivate$.outboundSchema = exports.LicenseKeyActivate$outboundSchema;
})(LicenseKeyActivate$ || (exports.LicenseKeyActivate$ = LicenseKeyActivate$ = {}));
function licenseKeyActivateToJSON(licenseKeyActivate) {
    return JSON.stringify(exports.LicenseKeyActivate$outboundSchema.parse(licenseKeyActivate));
}
function licenseKeyActivateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LicenseKeyActivate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LicenseKeyActivate' from JSON`);
}
