// @polar-sh/sdk@0.18.0/models/components/licensekeyvalidate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicenseKeyValidate$ = exports.LicenseKeyValidate$outboundSchema = exports.LicenseKeyValidate$inboundSchema = exports.Conditions$ = exports.Conditions$outboundSchema = exports.Conditions$inboundSchema = void 0;
exports.conditionsToJSON = conditionsToJSON;
exports.conditionsFromJSON = conditionsFromJSON;
exports.licenseKeyValidateToJSON = licenseKeyValidateToJSON;
exports.licenseKeyValidateFromJSON = licenseKeyValidateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.Conditions$inboundSchema = z.object({});
/** @internal */
exports.Conditions$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Conditions$;
(function (Conditions$) {
    /** @deprecated use `Conditions$inboundSchema` instead. */
    Conditions$.inboundSchema = exports.Conditions$inboundSchema;
    /** @deprecated use `Conditions$outboundSchema` instead. */
    Conditions$.outboundSchema = exports.Conditions$outboundSchema;
})(Conditions$ || (exports.Conditions$ = Conditions$ = {}));
function conditionsToJSON(conditions) {
    return JSON.stringify(exports.Conditions$outboundSchema.parse(conditions));
}
function conditionsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Conditions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Conditions' from JSON`);
}
/** @internal */
exports.LicenseKeyValidate$inboundSchema = z.object({
    key: z.string(),
    organization_id: z.string(),
    activation_id: z.nullable(z.string()).optional(),
    benefit_id: z.nullable(z.string()).optional(),
    user_id: z.nullable(z.string()).optional(),
    increment_usage: z.nullable(z.number().int()).optional(),
    conditions: z.lazy(() => exports.Conditions$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "organization_id": "organizationId",
        "activation_id": "activationId",
        "benefit_id": "benefitId",
        "user_id": "userId",
        "increment_usage": "incrementUsage",
    });
});
/** @internal */
exports.LicenseKeyValidate$outboundSchema = z.object({
    key: z.string(),
    organizationId: z.string(),
    activationId: z.nullable(z.string()).optional(),
    benefitId: z.nullable(z.string()).optional(),
    userId: z.nullable(z.string()).optional(),
    incrementUsage: z.nullable(z.number().int()).optional(),
    conditions: z.lazy(() => exports.Conditions$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        organizationId: "organization_id",
        activationId: "activation_id",
        benefitId: "benefit_id",
        userId: "user_id",
        incrementUsage: "increment_usage",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LicenseKeyValidate$;
(function (LicenseKeyValidate$) {
    /** @deprecated use `LicenseKeyValidate$inboundSchema` instead. */
    LicenseKeyValidate$.inboundSchema = exports.LicenseKeyValidate$inboundSchema;
    /** @deprecated use `LicenseKeyValidate$outboundSchema` instead. */
    LicenseKeyValidate$.outboundSchema = exports.LicenseKeyValidate$outboundSchema;
})(LicenseKeyValidate$ || (exports.LicenseKeyValidate$ = LicenseKeyValidate$ = {}));
function licenseKeyValidateToJSON(licenseKeyValidate) {
    return JSON.stringify(exports.LicenseKeyValidate$outboundSchema.parse(licenseKeyValidate));
}
function licenseKeyValidateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LicenseKeyValidate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LicenseKeyValidate' from JSON`);
}
