// @polar-sh/sdk@0.18.0/models/components/listresourceusersubscription.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListResourceUserSubscription$ = exports.ListResourceUserSubscription$outboundSchema = exports.ListResourceUserSubscription$inboundSchema = void 0;
exports.listResourceUserSubscriptionToJSON = listResourceUserSubscriptionToJSON;
exports.listResourceUserSubscriptionFromJSON = listResourceUserSubscriptionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const pagination_js_1 = require("./pagination.js");
const usersubscription_js_1 = require("./usersubscription.js");
/** @internal */
exports.ListResourceUserSubscription$inboundSchema = z.object({
    items: z.array(usersubscription_js_1.UserSubscription$inboundSchema),
    pagination: pagination_js_1.Pagination$inboundSchema,
});
/** @internal */
exports.ListResourceUserSubscription$outboundSchema = z.object({
    items: z.array(usersubscription_js_1.UserSubscription$outboundSchema),
    pagination: pagination_js_1.Pagination$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListResourceUserSubscription$;
(function (ListResourceUserSubscription$) {
    /** @deprecated use `ListResourceUserSubscription$inboundSchema` instead. */
    ListResourceUserSubscription$.inboundSchema = exports.ListResourceUserSubscription$inboundSchema;
    /** @deprecated use `ListResourceUserSubscription$outboundSchema` instead. */
    ListResourceUserSubscription$.outboundSchema = exports.ListResourceUserSubscription$outboundSchema;
})(ListResourceUserSubscription$ || (exports.ListResourceUserSubscription$ = ListResourceUserSubscription$ = {}));
function listResourceUserSubscriptionToJSON(listResourceUserSubscription) {
    return JSON.stringify(exports.ListResourceUserSubscription$outboundSchema.parse(listResourceUserSubscription));
}
function listResourceUserSubscriptionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListResourceUserSubscription$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListResourceUserSubscription' from JSON`);
}
