// @polar-sh/sdk@0.18.0/models/components/organizationcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationCreate$ = exports.OrganizationCreate$outboundSchema = exports.OrganizationCreate$inboundSchema = void 0;
exports.organizationCreateToJSON = organizationCreateToJSON;
exports.organizationCreateFromJSON = organizationCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const organizationfeaturesettings_js_1 = require("./organizationfeaturesettings.js");
/** @internal */
exports.OrganizationCreate$inboundSchema = z.object({
    name: z.string(),
    slug: z.string(),
    avatar_url: z.nullable(z.string()).optional(),
    feature_settings: z.nullable(organizationfeaturesettings_js_1.OrganizationFeatureSettings$inboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "avatar_url": "avatarUrl",
        "feature_settings": "featureSettings",
    });
});
/** @internal */
exports.OrganizationCreate$outboundSchema = z.object({
    name: z.string(),
    slug: z.string(),
    avatarUrl: z.nullable(z.string()).optional(),
    featureSettings: z.nullable(organizationfeaturesettings_js_1.OrganizationFeatureSettings$outboundSchema)
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        avatarUrl: "avatar_url",
        featureSettings: "feature_settings",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationCreate$;
(function (OrganizationCreate$) {
    /** @deprecated use `OrganizationCreate$inboundSchema` instead. */
    OrganizationCreate$.inboundSchema = exports.OrganizationCreate$inboundSchema;
    /** @deprecated use `OrganizationCreate$outboundSchema` instead. */
    OrganizationCreate$.outboundSchema = exports.OrganizationCreate$outboundSchema;
})(OrganizationCreate$ || (exports.OrganizationCreate$ = OrganizationCreate$ = {}));
function organizationCreateToJSON(organizationCreate) {
    return JSON.stringify(exports.OrganizationCreate$outboundSchema.parse(organizationCreate));
}
function organizationCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrganizationCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationCreate' from JSON`);
}
