// @polar-sh/sdk@0.18.0/models/components/productonetimecreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductOneTimeCreate$ = exports.ProductOneTimeCreate$outboundSchema = exports.ProductOneTimeCreate$inboundSchema = exports.Prices$ = exports.Prices$outboundSchema = exports.Prices$inboundSchema = exports.ProductOneTimeCreateMetadata$ = exports.ProductOneTimeCreateMetadata$outboundSchema = exports.ProductOneTimeCreateMetadata$inboundSchema = void 0;
exports.productOneTimeCreateMetadataToJSON = productOneTimeCreateMetadataToJSON;
exports.productOneTimeCreateMetadataFromJSON = productOneTimeCreateMetadataFromJSON;
exports.pricesToJSON = pricesToJSON;
exports.pricesFromJSON = pricesFromJSON;
exports.productOneTimeCreateToJSON = productOneTimeCreateToJSON;
exports.productOneTimeCreateFromJSON = productOneTimeCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const attachedcustomfieldcreate_js_1 = require("./attachedcustomfieldcreate.js");
const productpriceonetimecustomcreate_js_1 = require("./productpriceonetimecustomcreate.js");
const productpriceonetimefixedcreate_js_1 = require("./productpriceonetimefixedcreate.js");
const productpriceonetimefreecreate_js_1 = require("./productpriceonetimefreecreate.js");
/** @internal */
exports.ProductOneTimeCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
exports.ProductOneTimeCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductOneTimeCreateMetadata$;
(function (ProductOneTimeCreateMetadata$) {
    /** @deprecated use `ProductOneTimeCreateMetadata$inboundSchema` instead. */
    ProductOneTimeCreateMetadata$.inboundSchema = exports.ProductOneTimeCreateMetadata$inboundSchema;
    /** @deprecated use `ProductOneTimeCreateMetadata$outboundSchema` instead. */
    ProductOneTimeCreateMetadata$.outboundSchema = exports.ProductOneTimeCreateMetadata$outboundSchema;
})(ProductOneTimeCreateMetadata$ || (exports.ProductOneTimeCreateMetadata$ = ProductOneTimeCreateMetadata$ = {}));
function productOneTimeCreateMetadataToJSON(productOneTimeCreateMetadata) {
    return JSON.stringify(exports.ProductOneTimeCreateMetadata$outboundSchema.parse(productOneTimeCreateMetadata));
}
function productOneTimeCreateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductOneTimeCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductOneTimeCreateMetadata' from JSON`);
}
/** @internal */
exports.Prices$inboundSchema = z
    .union([
    productpriceonetimefreecreate_js_1.ProductPriceOneTimeFreeCreate$inboundSchema,
    productpriceonetimefixedcreate_js_1.ProductPriceOneTimeFixedCreate$inboundSchema,
    productpriceonetimecustomcreate_js_1.ProductPriceOneTimeCustomCreate$inboundSchema,
]);
/** @internal */
exports.Prices$outboundSchema = z.union([
    productpriceonetimefreecreate_js_1.ProductPriceOneTimeFreeCreate$outboundSchema,
    productpriceonetimefixedcreate_js_1.ProductPriceOneTimeFixedCreate$outboundSchema,
    productpriceonetimecustomcreate_js_1.ProductPriceOneTimeCustomCreate$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Prices$;
(function (Prices$) {
    /** @deprecated use `Prices$inboundSchema` instead. */
    Prices$.inboundSchema = exports.Prices$inboundSchema;
    /** @deprecated use `Prices$outboundSchema` instead. */
    Prices$.outboundSchema = exports.Prices$outboundSchema;
})(Prices$ || (exports.Prices$ = Prices$ = {}));
function pricesToJSON(prices) {
    return JSON.stringify(exports.Prices$outboundSchema.parse(prices));
}
function pricesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Prices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Prices' from JSON`);
}
/** @internal */
exports.ProductOneTimeCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.array(z.union([
        productpriceonetimefreecreate_js_1.ProductPriceOneTimeFreeCreate$inboundSchema,
        productpriceonetimefixedcreate_js_1.ProductPriceOneTimeFixedCreate$inboundSchema,
        productpriceonetimecustomcreate_js_1.ProductPriceOneTimeCustomCreate$inboundSchema,
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attached_custom_fields: z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$inboundSchema)
        .optional(),
    organization_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "attached_custom_fields": "attachedCustomFields",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.ProductOneTimeCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.array(z.union([
        productpriceonetimefreecreate_js_1.ProductPriceOneTimeFreeCreate$outboundSchema,
        productpriceonetimefixedcreate_js_1.ProductPriceOneTimeFixedCreate$outboundSchema,
        productpriceonetimecustomcreate_js_1.ProductPriceOneTimeCustomCreate$outboundSchema,
    ])),
    medias: z.nullable(z.array(z.string())).optional(),
    attachedCustomFields: z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$outboundSchema)
        .optional(),
    organizationId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        attachedCustomFields: "attached_custom_fields",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductOneTimeCreate$;
(function (ProductOneTimeCreate$) {
    /** @deprecated use `ProductOneTimeCreate$inboundSchema` instead. */
    ProductOneTimeCreate$.inboundSchema = exports.ProductOneTimeCreate$inboundSchema;
    /** @deprecated use `ProductOneTimeCreate$outboundSchema` instead. */
    ProductOneTimeCreate$.outboundSchema = exports.ProductOneTimeCreate$outboundSchema;
})(ProductOneTimeCreate$ || (exports.ProductOneTimeCreate$ = ProductOneTimeCreate$ = {}));
function productOneTimeCreateToJSON(productOneTimeCreate) {
    return JSON.stringify(exports.ProductOneTimeCreate$outboundSchema.parse(productOneTimeCreate));
}
function productOneTimeCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductOneTimeCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductOneTimeCreate' from JSON`);
}
