// @polar-sh/sdk@0.18.0/models/components/product.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Product$ = exports.Product$outboundSchema = exports.Product$inboundSchema = exports.ProductMetadata$ = exports.ProductMetadata$outboundSchema = exports.ProductMetadata$inboundSchema = void 0;
exports.productMetadataToJSON = productMetadataToJSON;
exports.productMetadataFromJSON = productMetadataFromJSON;
exports.productToJSON = productToJSON;
exports.productFromJSON = productFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const attachedcustomfield_js_1 = require("./attachedcustomfield.js");
const benefit_js_1 = require("./benefit.js");
const productmediafileread_js_1 = require("./productmediafileread.js");
const productprice_js_1 = require("./productprice.js");
/** @internal */
exports.ProductMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
exports.ProductMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductMetadata$;
(function (ProductMetadata$) {
    /** @deprecated use `ProductMetadata$inboundSchema` instead. */
    ProductMetadata$.inboundSchema = exports.ProductMetadata$inboundSchema;
    /** @deprecated use `ProductMetadata$outboundSchema` instead. */
    ProductMetadata$.outboundSchema = exports.ProductMetadata$outboundSchema;
})(ProductMetadata$ || (exports.ProductMetadata$ = ProductMetadata$ = {}));
function productMetadataToJSON(productMetadata) {
    return JSON.stringify(exports.ProductMetadata$outboundSchema.parse(productMetadata));
}
function productMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductMetadata' from JSON`);
}
/** @internal */
exports.Product$inboundSchema = z.object({
    created_at: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    modified_at: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    is_recurring: z.boolean(),
    is_archived: z.boolean(),
    organization_id: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()])),
    prices: z.array(productprice_js_1.ProductPrice$inboundSchema),
    benefits: z.array(benefit_js_1.Benefit$inboundSchema),
    medias: z.array(productmediafileread_js_1.ProductMediaFileRead$inboundSchema),
    attached_custom_fields: z.array(attachedcustomfield_js_1.AttachedCustomField$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "is_recurring": "isRecurring",
        "is_archived": "isArchived",
        "organization_id": "organizationId",
        "attached_custom_fields": "attachedCustomFields",
    });
});
/** @internal */
exports.Product$outboundSchema = z.object({
    createdAt: z.date().transform(v => v.toISOString()),
    modifiedAt: z.nullable(z.date().transform(v => v.toISOString())),
    id: z.string(),
    name: z.string(),
    description: z.nullable(z.string()),
    isRecurring: z.boolean(),
    isArchived: z.boolean(),
    organizationId: z.string(),
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()])),
    prices: z.array(productprice_js_1.ProductPrice$outboundSchema),
    benefits: z.array(benefit_js_1.Benefit$outboundSchema),
    medias: z.array(productmediafileread_js_1.ProductMediaFileRead$outboundSchema),
    attachedCustomFields: z.array(attachedcustomfield_js_1.AttachedCustomField$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        createdAt: "created_at",
        modifiedAt: "modified_at",
        isRecurring: "is_recurring",
        isArchived: "is_archived",
        organizationId: "organization_id",
        attachedCustomFields: "attached_custom_fields",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Product$;
(function (Product$) {
    /** @deprecated use `Product$inboundSchema` instead. */
    Product$.inboundSchema = exports.Product$inboundSchema;
    /** @deprecated use `Product$outboundSchema` instead. */
    Product$.outboundSchema = exports.Product$outboundSchema;
})(Product$ || (exports.Product$ = Product$ = {}));
function productToJSON(product) {
    return JSON.stringify(exports.Product$outboundSchema.parse(product));
}
function productFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Product$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Product' from JSON`);
}
