// @polar-sh/sdk@0.18.0/models/components/productrecurringcreate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductRecurringCreate$ = exports.ProductRecurringCreate$outboundSchema = exports.ProductRecurringCreate$inboundSchema = exports.ProductRecurringCreatePrices$ = exports.ProductRecurringCreatePrices$outboundSchema = exports.ProductRecurringCreatePrices$inboundSchema = exports.ProductRecurringCreateMetadata$ = exports.ProductRecurringCreateMetadata$outboundSchema = exports.ProductRecurringCreateMetadata$inboundSchema = void 0;
exports.productRecurringCreateMetadataToJSON = productRecurringCreateMetadataToJSON;
exports.productRecurringCreateMetadataFromJSON = productRecurringCreateMetadataFromJSON;
exports.productRecurringCreatePricesToJSON = productRecurringCreatePricesToJSON;
exports.productRecurringCreatePricesFromJSON = productRecurringCreatePricesFromJSON;
exports.productRecurringCreateToJSON = productRecurringCreateToJSON;
exports.productRecurringCreateFromJSON = productRecurringCreateFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const attachedcustomfieldcreate_js_1 = require("./attachedcustomfieldcreate.js");
const productpricerecurringfixedcreate_js_1 = require("./productpricerecurringfixedcreate.js");
const productpricerecurringfreecreate_js_1 = require("./productpricerecurringfreecreate.js");
/** @internal */
exports.ProductRecurringCreateMetadata$inboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/** @internal */
exports.ProductRecurringCreateMetadata$outboundSchema = z.union([z.string(), z.number().int(), z.boolean()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductRecurringCreateMetadata$;
(function (ProductRecurringCreateMetadata$) {
    /** @deprecated use `ProductRecurringCreateMetadata$inboundSchema` instead. */
    ProductRecurringCreateMetadata$.inboundSchema = exports.ProductRecurringCreateMetadata$inboundSchema;
    /** @deprecated use `ProductRecurringCreateMetadata$outboundSchema` instead. */
    ProductRecurringCreateMetadata$.outboundSchema = exports.ProductRecurringCreateMetadata$outboundSchema;
})(ProductRecurringCreateMetadata$ || (exports.ProductRecurringCreateMetadata$ = ProductRecurringCreateMetadata$ = {}));
function productRecurringCreateMetadataToJSON(productRecurringCreateMetadata) {
    return JSON.stringify(exports.ProductRecurringCreateMetadata$outboundSchema.parse(productRecurringCreateMetadata));
}
function productRecurringCreateMetadataFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductRecurringCreateMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductRecurringCreateMetadata' from JSON`);
}
/** @internal */
exports.ProductRecurringCreatePrices$inboundSchema = z.union([
    z.array(productpricerecurringfixedcreate_js_1.ProductPriceRecurringFixedCreate$inboundSchema),
    z.array(productpricerecurringfreecreate_js_1.ProductPriceRecurringFreeCreate$inboundSchema),
]);
/** @internal */
exports.ProductRecurringCreatePrices$outboundSchema = z.union([
    z.array(productpricerecurringfixedcreate_js_1.ProductPriceRecurringFixedCreate$outboundSchema),
    z.array(productpricerecurringfreecreate_js_1.ProductPriceRecurringFreeCreate$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductRecurringCreatePrices$;
(function (ProductRecurringCreatePrices$) {
    /** @deprecated use `ProductRecurringCreatePrices$inboundSchema` instead. */
    ProductRecurringCreatePrices$.inboundSchema = exports.ProductRecurringCreatePrices$inboundSchema;
    /** @deprecated use `ProductRecurringCreatePrices$outboundSchema` instead. */
    ProductRecurringCreatePrices$.outboundSchema = exports.ProductRecurringCreatePrices$outboundSchema;
})(ProductRecurringCreatePrices$ || (exports.ProductRecurringCreatePrices$ = ProductRecurringCreatePrices$ = {}));
function productRecurringCreatePricesToJSON(productRecurringCreatePrices) {
    return JSON.stringify(exports.ProductRecurringCreatePrices$outboundSchema.parse(productRecurringCreatePrices));
}
function productRecurringCreatePricesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductRecurringCreatePrices$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductRecurringCreatePrices' from JSON`);
}
/** @internal */
exports.ProductRecurringCreate$inboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.union([
        z.array(productpricerecurringfixedcreate_js_1.ProductPriceRecurringFixedCreate$inboundSchema),
        z.array(productpricerecurringfreecreate_js_1.ProductPriceRecurringFreeCreate$inboundSchema),
    ]),
    medias: z.nullable(z.array(z.string())).optional(),
    attached_custom_fields: z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$inboundSchema)
        .optional(),
    organization_id: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "attached_custom_fields": "attachedCustomFields",
        "organization_id": "organizationId",
    });
});
/** @internal */
exports.ProductRecurringCreate$outboundSchema = z.object({
    metadata: z.record(z.union([z.string(), z.number().int(), z.boolean()]))
        .optional(),
    name: z.string(),
    description: z.nullable(z.string()).optional(),
    prices: z.union([
        z.array(productpricerecurringfixedcreate_js_1.ProductPriceRecurringFixedCreate$outboundSchema),
        z.array(productpricerecurringfreecreate_js_1.ProductPriceRecurringFreeCreate$outboundSchema),
    ]),
    medias: z.nullable(z.array(z.string())).optional(),
    attachedCustomFields: z.array(attachedcustomfieldcreate_js_1.AttachedCustomFieldCreate$outboundSchema)
        .optional(),
    organizationId: z.nullable(z.string()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        attachedCustomFields: "attached_custom_fields",
        organizationId: "organization_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ProductRecurringCreate$;
(function (ProductRecurringCreate$) {
    /** @deprecated use `ProductRecurringCreate$inboundSchema` instead. */
    ProductRecurringCreate$.inboundSchema = exports.ProductRecurringCreate$inboundSchema;
    /** @deprecated use `ProductRecurringCreate$outboundSchema` instead. */
    ProductRecurringCreate$.outboundSchema = exports.ProductRecurringCreate$outboundSchema;
})(ProductRecurringCreate$ || (exports.ProductRecurringCreate$ = ProductRecurringCreate$ = {}));
function productRecurringCreateToJSON(productRecurringCreate) {
    return JSON.stringify(exports.ProductRecurringCreate$outboundSchema.parse(productRecurringCreate));
}
function productRecurringCreateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ProductRecurringCreate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProductRecurringCreate' from JSON`);
}
