// @polar-sh/sdk@0.18.0/models/components/scope.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Scope$ = exports.Scope$outboundSchema = exports.Scope$inboundSchema = exports.Scope = void 0;
const z = __importStar(require("zod"));
exports.Scope = {
    Openid: "openid",
    Profile: "profile",
    Email: "email",
    UserRead: "user:read",
    Admin: "admin",
    WebDefault: "web_default",
    OrganizationsRead: "organizations:read",
    OrganizationsWrite: "organizations:write",
    CustomFieldsRead: "custom_fields:read",
    CustomFieldsWrite: "custom_fields:write",
    DiscountsRead: "discounts:read",
    DiscountsWrite: "discounts:write",
    CheckoutLinksRead: "checkout_links:read",
    CheckoutLinksWrite: "checkout_links:write",
    CheckoutsRead: "checkouts:read",
    CheckoutsWrite: "checkouts:write",
    ProductsRead: "products:read",
    ProductsWrite: "products:write",
    BenefitsRead: "benefits:read",
    BenefitsWrite: "benefits:write",
    FilesRead: "files:read",
    FilesWrite: "files:write",
    SubscriptionsRead: "subscriptions:read",
    SubscriptionsWrite: "subscriptions:write",
    OrdersRead: "orders:read",
    MetricsRead: "metrics:read",
    WebhooksRead: "webhooks:read",
    WebhooksWrite: "webhooks:write",
    ExternalOrganizationsRead: "external_organizations:read",
    LicenseKeysRead: "license_keys:read",
    LicenseKeysWrite: "license_keys:write",
    RepositoriesRead: "repositories:read",
    RepositoriesWrite: "repositories:write",
    IssuesRead: "issues:read",
    IssuesWrite: "issues:write",
    UserBenefitsRead: "user:benefits:read",
    UserOrdersRead: "user:orders:read",
    UserSubscriptionsRead: "user:subscriptions:read",
    UserSubscriptionsWrite: "user:subscriptions:write",
    UserDownloadablesRead: "user:downloadables:read",
    UserLicenseKeysRead: "user:license_keys:read",
    UserAdvertisementCampaignsRead: "user:advertisement_campaigns:read",
    UserAdvertisementCampaignsWrite: "user:advertisement_campaigns:write",
};
/** @internal */
exports.Scope$inboundSchema = z.nativeEnum(exports.Scope);
/** @internal */
exports.Scope$outboundSchema = exports.Scope$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Scope$;
(function (Scope$) {
    /** @deprecated use `Scope$inboundSchema` instead. */
    Scope$.inboundSchema = exports.Scope$inboundSchema;
    /** @deprecated use `Scope$outboundSchema` instead. */
    Scope$.outboundSchema = exports.Scope$outboundSchema;
})(Scope$ || (exports.Scope$ = Scope$ = {}));
