// @polar-sh/sdk@0.18.0/models/components/webhooksubscriptionupdatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookSubscriptionUpdatedPayload$ = exports.WebhookSubscriptionUpdatedPayload$outboundSchema = exports.WebhookSubscriptionUpdatedPayload$inboundSchema = exports.WebhookSubscriptionUpdatedPayloadType$ = exports.WebhookSubscriptionUpdatedPayloadType$outboundSchema = exports.WebhookSubscriptionUpdatedPayloadType$inboundSchema = exports.WebhookSubscriptionUpdatedPayloadType = void 0;
exports.webhookSubscriptionUpdatedPayloadToJSON = webhookSubscriptionUpdatedPayloadToJSON;
exports.webhookSubscriptionUpdatedPayloadFromJSON = webhookSubscriptionUpdatedPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const subscription_js_1 = require("./subscription.js");
exports.WebhookSubscriptionUpdatedPayloadType = {
    SubscriptionUpdated: "subscription.updated",
};
/** @internal */
exports.WebhookSubscriptionUpdatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookSubscriptionUpdatedPayloadType);
/** @internal */
exports.WebhookSubscriptionUpdatedPayloadType$outboundSchema = exports.WebhookSubscriptionUpdatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookSubscriptionUpdatedPayloadType$;
(function (WebhookSubscriptionUpdatedPayloadType$) {
    /** @deprecated use `WebhookSubscriptionUpdatedPayloadType$inboundSchema` instead. */
    WebhookSubscriptionUpdatedPayloadType$.inboundSchema = exports.WebhookSubscriptionUpdatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookSubscriptionUpdatedPayloadType$outboundSchema` instead. */
    WebhookSubscriptionUpdatedPayloadType$.outboundSchema = exports.WebhookSubscriptionUpdatedPayloadType$outboundSchema;
})(WebhookSubscriptionUpdatedPayloadType$ || (exports.WebhookSubscriptionUpdatedPayloadType$ = WebhookSubscriptionUpdatedPayloadType$ = {}));
/** @internal */
exports.WebhookSubscriptionUpdatedPayload$inboundSchema = z.object({
    type: z.literal("subscription.updated").optional(),
    data: subscription_js_1.Subscription$inboundSchema,
});
/** @internal */
exports.WebhookSubscriptionUpdatedPayload$outboundSchema = z.object({
    type: z.literal("subscription.updated").default("subscription.updated"),
    data: subscription_js_1.Subscription$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookSubscriptionUpdatedPayload$;
(function (WebhookSubscriptionUpdatedPayload$) {
    /** @deprecated use `WebhookSubscriptionUpdatedPayload$inboundSchema` instead. */
    WebhookSubscriptionUpdatedPayload$.inboundSchema = exports.WebhookSubscriptionUpdatedPayload$inboundSchema;
    /** @deprecated use `WebhookSubscriptionUpdatedPayload$outboundSchema` instead. */
    WebhookSubscriptionUpdatedPayload$.outboundSchema = exports.WebhookSubscriptionUpdatedPayload$outboundSchema;
})(WebhookSubscriptionUpdatedPayload$ || (exports.WebhookSubscriptionUpdatedPayload$ = WebhookSubscriptionUpdatedPayload$ = {}));
function webhookSubscriptionUpdatedPayloadToJSON(webhookSubscriptionUpdatedPayload) {
    return JSON.stringify(exports.WebhookSubscriptionUpdatedPayload$outboundSchema.parse(webhookSubscriptionUpdatedPayload));
}
function webhookSubscriptionUpdatedPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookSubscriptionUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookSubscriptionUpdatedPayload' from JSON`);
}
