// @polar-sh/sdk@0.18.0/models/components/webhooksubscriptioncreatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookSubscriptionCreatedPayload$ = exports.WebhookSubscriptionCreatedPayload$outboundSchema = exports.WebhookSubscriptionCreatedPayload$inboundSchema = exports.WebhookSubscriptionCreatedPayloadType$ = exports.WebhookSubscriptionCreatedPayloadType$outboundSchema = exports.WebhookSubscriptionCreatedPayloadType$inboundSchema = exports.WebhookSubscriptionCreatedPayloadType = void 0;
exports.webhookSubscriptionCreatedPayloadToJSON = webhookSubscriptionCreatedPayloadToJSON;
exports.webhookSubscriptionCreatedPayloadFromJSON = webhookSubscriptionCreatedPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const subscription_js_1 = require("./subscription.js");
exports.WebhookSubscriptionCreatedPayloadType = {
    SubscriptionCreated: "subscription.created",
};
/** @internal */
exports.WebhookSubscriptionCreatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookSubscriptionCreatedPayloadType);
/** @internal */
exports.WebhookSubscriptionCreatedPayloadType$outboundSchema = exports.WebhookSubscriptionCreatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookSubscriptionCreatedPayloadType$;
(function (WebhookSubscriptionCreatedPayloadType$) {
    /** @deprecated use `WebhookSubscriptionCreatedPayloadType$inboundSchema` instead. */
    WebhookSubscriptionCreatedPayloadType$.inboundSchema = exports.WebhookSubscriptionCreatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookSubscriptionCreatedPayloadType$outboundSchema` instead. */
    WebhookSubscriptionCreatedPayloadType$.outboundSchema = exports.WebhookSubscriptionCreatedPayloadType$outboundSchema;
})(WebhookSubscriptionCreatedPayloadType$ || (exports.WebhookSubscriptionCreatedPayloadType$ = WebhookSubscriptionCreatedPayloadType$ = {}));
/** @internal */
exports.WebhookSubscriptionCreatedPayload$inboundSchema = z.object({
    type: z.literal("subscription.created").optional(),
    data: subscription_js_1.Subscription$inboundSchema,
});
/** @internal */
exports.WebhookSubscriptionCreatedPayload$outboundSchema = z.object({
    type: z.literal("subscription.created").default("subscription.created"),
    data: subscription_js_1.Subscription$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookSubscriptionCreatedPayload$;
(function (WebhookSubscriptionCreatedPayload$) {
    /** @deprecated use `WebhookSubscriptionCreatedPayload$inboundSchema` instead. */
    WebhookSubscriptionCreatedPayload$.inboundSchema = exports.WebhookSubscriptionCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookSubscriptionCreatedPayload$outboundSchema` instead. */
    WebhookSubscriptionCreatedPayload$.outboundSchema = exports.WebhookSubscriptionCreatedPayload$outboundSchema;
})(WebhookSubscriptionCreatedPayload$ || (exports.WebhookSubscriptionCreatedPayload$ = WebhookSubscriptionCreatedPayload$ = {}));
function webhookSubscriptionCreatedPayloadToJSON(webhookSubscriptionCreatedPayload) {
    return JSON.stringify(exports.WebhookSubscriptionCreatedPayload$outboundSchema.parse(webhookSubscriptionCreatedPayload));
}
function webhookSubscriptionCreatedPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookSubscriptionCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookSubscriptionCreatedPayload' from JSON`);
}
