// @polar-sh/sdk@0.18.0/models/components/webhookcheckoutcreatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookCheckoutCreatedPayload$ = exports.WebhookCheckoutCreatedPayload$outboundSchema = exports.WebhookCheckoutCreatedPayload$inboundSchema = exports.WebhookCheckoutCreatedPayloadType$ = exports.WebhookCheckoutCreatedPayloadType$outboundSchema = exports.WebhookCheckoutCreatedPayloadType$inboundSchema = exports.WebhookCheckoutCreatedPayloadType = void 0;
exports.webhookCheckoutCreatedPayloadToJSON = webhookCheckoutCreatedPayloadToJSON;
exports.webhookCheckoutCreatedPayloadFromJSON = webhookCheckoutCreatedPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const checkout_js_1 = require("./checkout.js");
exports.WebhookCheckoutCreatedPayloadType = {
    CheckoutCreated: "checkout.created",
};
/** @internal */
exports.WebhookCheckoutCreatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookCheckoutCreatedPayloadType);
/** @internal */
exports.WebhookCheckoutCreatedPayloadType$outboundSchema = exports.WebhookCheckoutCreatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookCheckoutCreatedPayloadType$;
(function (WebhookCheckoutCreatedPayloadType$) {
    /** @deprecated use `WebhookCheckoutCreatedPayloadType$inboundSchema` instead. */
    WebhookCheckoutCreatedPayloadType$.inboundSchema = exports.WebhookCheckoutCreatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookCheckoutCreatedPayloadType$outboundSchema` instead. */
    WebhookCheckoutCreatedPayloadType$.outboundSchema = exports.WebhookCheckoutCreatedPayloadType$outboundSchema;
})(WebhookCheckoutCreatedPayloadType$ || (exports.WebhookCheckoutCreatedPayloadType$ = WebhookCheckoutCreatedPayloadType$ = {}));
/** @internal */
exports.WebhookCheckoutCreatedPayload$inboundSchema = z.object({
    type: z.literal("checkout.created").optional(),
    data: checkout_js_1.Checkout$inboundSchema,
});
/** @internal */
exports.WebhookCheckoutCreatedPayload$outboundSchema = z.object({
    type: z.literal("checkout.created").default("checkout.created"),
    data: checkout_js_1.Checkout$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookCheckoutCreatedPayload$;
(function (WebhookCheckoutCreatedPayload$) {
    /** @deprecated use `WebhookCheckoutCreatedPayload$inboundSchema` instead. */
    WebhookCheckoutCreatedPayload$.inboundSchema = exports.WebhookCheckoutCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookCheckoutCreatedPayload$outboundSchema` instead. */
    WebhookCheckoutCreatedPayload$.outboundSchema = exports.WebhookCheckoutCreatedPayload$outboundSchema;
})(WebhookCheckoutCreatedPayload$ || (exports.WebhookCheckoutCreatedPayload$ = WebhookCheckoutCreatedPayload$ = {}));
function webhookCheckoutCreatedPayloadToJSON(webhookCheckoutCreatedPayload) {
    return JSON.stringify(exports.WebhookCheckoutCreatedPayload$outboundSchema.parse(webhookCheckoutCreatedPayload));
}
function webhookCheckoutCreatedPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookCheckoutCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCheckoutCreatedPayload' from JSON`);
}
