// @polar-sh/sdk@0.18.0/models/components/webhookcheckoutupdatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookCheckoutUpdatedPayload$ = exports.WebhookCheckoutUpdatedPayload$outboundSchema = exports.WebhookCheckoutUpdatedPayload$inboundSchema = exports.WebhookCheckoutUpdatedPayloadType$ = exports.WebhookCheckoutUpdatedPayloadType$outboundSchema = exports.WebhookCheckoutUpdatedPayloadType$inboundSchema = exports.WebhookCheckoutUpdatedPayloadType = void 0;
exports.webhookCheckoutUpdatedPayloadToJSON = webhookCheckoutUpdatedPayloadToJSON;
exports.webhookCheckoutUpdatedPayloadFromJSON = webhookCheckoutUpdatedPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const checkout_js_1 = require("./checkout.js");
exports.WebhookCheckoutUpdatedPayloadType = {
    CheckoutUpdated: "checkout.updated",
};
/** @internal */
exports.WebhookCheckoutUpdatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookCheckoutUpdatedPayloadType);
/** @internal */
exports.WebhookCheckoutUpdatedPayloadType$outboundSchema = exports.WebhookCheckoutUpdatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookCheckoutUpdatedPayloadType$;
(function (WebhookCheckoutUpdatedPayloadType$) {
    /** @deprecated use `WebhookCheckoutUpdatedPayloadType$inboundSchema` instead. */
    WebhookCheckoutUpdatedPayloadType$.inboundSchema = exports.WebhookCheckoutUpdatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookCheckoutUpdatedPayloadType$outboundSchema` instead. */
    WebhookCheckoutUpdatedPayloadType$.outboundSchema = exports.WebhookCheckoutUpdatedPayloadType$outboundSchema;
})(WebhookCheckoutUpdatedPayloadType$ || (exports.WebhookCheckoutUpdatedPayloadType$ = WebhookCheckoutUpdatedPayloadType$ = {}));
/** @internal */
exports.WebhookCheckoutUpdatedPayload$inboundSchema = z.object({
    type: z.literal("checkout.updated").optional(),
    data: checkout_js_1.Checkout$inboundSchema,
});
/** @internal */
exports.WebhookCheckoutUpdatedPayload$outboundSchema = z.object({
    type: z.literal("checkout.updated").default("checkout.updated"),
    data: checkout_js_1.Checkout$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookCheckoutUpdatedPayload$;
(function (WebhookCheckoutUpdatedPayload$) {
    /** @deprecated use `WebhookCheckoutUpdatedPayload$inboundSchema` instead. */
    WebhookCheckoutUpdatedPayload$.inboundSchema = exports.WebhookCheckoutUpdatedPayload$inboundSchema;
    /** @deprecated use `WebhookCheckoutUpdatedPayload$outboundSchema` instead. */
    WebhookCheckoutUpdatedPayload$.outboundSchema = exports.WebhookCheckoutUpdatedPayload$outboundSchema;
})(WebhookCheckoutUpdatedPayload$ || (exports.WebhookCheckoutUpdatedPayload$ = WebhookCheckoutUpdatedPayload$ = {}));
function webhookCheckoutUpdatedPayloadToJSON(webhookCheckoutUpdatedPayload) {
    return JSON.stringify(exports.WebhookCheckoutUpdatedPayload$outboundSchema.parse(webhookCheckoutUpdatedPayload));
}
function webhookCheckoutUpdatedPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookCheckoutUpdatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookCheckoutUpdatedPayload' from JSON`);
}
