// @polar-sh/sdk@0.18.0/models/components/webhookproductcreatedpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookProductCreatedPayload$ = exports.WebhookProductCreatedPayload$outboundSchema = exports.WebhookProductCreatedPayload$inboundSchema = exports.WebhookProductCreatedPayloadType$ = exports.WebhookProductCreatedPayloadType$outboundSchema = exports.WebhookProductCreatedPayloadType$inboundSchema = exports.WebhookProductCreatedPayloadType = void 0;
exports.webhookProductCreatedPayloadToJSON = webhookProductCreatedPayloadToJSON;
exports.webhookProductCreatedPayloadFromJSON = webhookProductCreatedPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const product_js_1 = require("./product.js");
exports.WebhookProductCreatedPayloadType = {
    ProductCreated: "product.created",
};
/** @internal */
exports.WebhookProductCreatedPayloadType$inboundSchema = z.nativeEnum(exports.WebhookProductCreatedPayloadType);
/** @internal */
exports.WebhookProductCreatedPayloadType$outboundSchema = exports.WebhookProductCreatedPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookProductCreatedPayloadType$;
(function (WebhookProductCreatedPayloadType$) {
    /** @deprecated use `WebhookProductCreatedPayloadType$inboundSchema` instead. */
    WebhookProductCreatedPayloadType$.inboundSchema = exports.WebhookProductCreatedPayloadType$inboundSchema;
    /** @deprecated use `WebhookProductCreatedPayloadType$outboundSchema` instead. */
    WebhookProductCreatedPayloadType$.outboundSchema = exports.WebhookProductCreatedPayloadType$outboundSchema;
})(WebhookProductCreatedPayloadType$ || (exports.WebhookProductCreatedPayloadType$ = WebhookProductCreatedPayloadType$ = {}));
/** @internal */
exports.WebhookProductCreatedPayload$inboundSchema = z.object({
    type: z.literal("product.created").optional(),
    data: product_js_1.Product$inboundSchema,
});
/** @internal */
exports.WebhookProductCreatedPayload$outboundSchema = z.object({
    type: z.literal("product.created").default("product.created"),
    data: product_js_1.Product$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookProductCreatedPayload$;
(function (WebhookProductCreatedPayload$) {
    /** @deprecated use `WebhookProductCreatedPayload$inboundSchema` instead. */
    WebhookProductCreatedPayload$.inboundSchema = exports.WebhookProductCreatedPayload$inboundSchema;
    /** @deprecated use `WebhookProductCreatedPayload$outboundSchema` instead. */
    WebhookProductCreatedPayload$.outboundSchema = exports.WebhookProductCreatedPayload$outboundSchema;
})(WebhookProductCreatedPayload$ || (exports.WebhookProductCreatedPayload$ = WebhookProductCreatedPayload$ = {}));
function webhookProductCreatedPayloadToJSON(webhookProductCreatedPayload) {
    return JSON.stringify(exports.WebhookProductCreatedPayload$outboundSchema.parse(webhookProductCreatedPayload));
}
function webhookProductCreatedPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookProductCreatedPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookProductCreatedPayload' from JSON`);
}
