// @polar-sh/sdk@0.18.0/models/components/webhooksubscriptioncanceledpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookSubscriptionCanceledPayload$ = exports.WebhookSubscriptionCanceledPayload$outboundSchema = exports.WebhookSubscriptionCanceledPayload$inboundSchema = exports.WebhookSubscriptionCanceledPayloadType$ = exports.WebhookSubscriptionCanceledPayloadType$outboundSchema = exports.WebhookSubscriptionCanceledPayloadType$inboundSchema = exports.WebhookSubscriptionCanceledPayloadType = void 0;
exports.webhookSubscriptionCanceledPayloadToJSON = webhookSubscriptionCanceledPayloadToJSON;
exports.webhookSubscriptionCanceledPayloadFromJSON = webhookSubscriptionCanceledPayloadFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const subscription_js_1 = require("./subscription.js");
exports.WebhookSubscriptionCanceledPayloadType = {
    SubscriptionCanceled: "subscription.canceled",
};
/** @internal */
exports.WebhookSubscriptionCanceledPayloadType$inboundSchema = z.nativeEnum(exports.WebhookSubscriptionCanceledPayloadType);
/** @internal */
exports.WebhookSubscriptionCanceledPayloadType$outboundSchema = exports.WebhookSubscriptionCanceledPayloadType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookSubscriptionCanceledPayloadType$;
(function (WebhookSubscriptionCanceledPayloadType$) {
    /** @deprecated use `WebhookSubscriptionCanceledPayloadType$inboundSchema` instead. */
    WebhookSubscriptionCanceledPayloadType$.inboundSchema = exports.WebhookSubscriptionCanceledPayloadType$inboundSchema;
    /** @deprecated use `WebhookSubscriptionCanceledPayloadType$outboundSchema` instead. */
    WebhookSubscriptionCanceledPayloadType$.outboundSchema = exports.WebhookSubscriptionCanceledPayloadType$outboundSchema;
})(WebhookSubscriptionCanceledPayloadType$ || (exports.WebhookSubscriptionCanceledPayloadType$ = WebhookSubscriptionCanceledPayloadType$ = {}));
/** @internal */
exports.WebhookSubscriptionCanceledPayload$inboundSchema = z.object({
    type: z.literal("subscription.canceled").optional(),
    data: subscription_js_1.Subscription$inboundSchema,
});
/** @internal */
exports.WebhookSubscriptionCanceledPayload$outboundSchema = z.object({
    type: z.literal("subscription.canceled").default("subscription.canceled"),
    data: subscription_js_1.Subscription$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookSubscriptionCanceledPayload$;
(function (WebhookSubscriptionCanceledPayload$) {
    /** @deprecated use `WebhookSubscriptionCanceledPayload$inboundSchema` instead. */
    WebhookSubscriptionCanceledPayload$.inboundSchema = exports.WebhookSubscriptionCanceledPayload$inboundSchema;
    /** @deprecated use `WebhookSubscriptionCanceledPayload$outboundSchema` instead. */
    WebhookSubscriptionCanceledPayload$.outboundSchema = exports.WebhookSubscriptionCanceledPayload$outboundSchema;
})(WebhookSubscriptionCanceledPayload$ || (exports.WebhookSubscriptionCanceledPayload$ = WebhookSubscriptionCanceledPayload$ = {}));
function webhookSubscriptionCanceledPayloadToJSON(webhookSubscriptionCanceledPayload) {
    return JSON.stringify(exports.WebhookSubscriptionCanceledPayload$outboundSchema.parse(webhookSubscriptionCanceledPayload));
}
function webhookSubscriptionCanceledPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookSubscriptionCanceledPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookSubscriptionCanceledPayload' from JSON`);
}
