// @polar-sh/sdk@0.18.0/models/operations/customfieldsget.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomFieldsGetRequest$ = exports.CustomFieldsGetRequest$outboundSchema = exports.CustomFieldsGetRequest$inboundSchema = void 0;
exports.customFieldsGetRequestToJSON = customFieldsGetRequestToJSON;
exports.customFieldsGetRequestFromJSON = customFieldsGetRequestFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomFieldsGetRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.CustomFieldsGetRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomFieldsGetRequest$;
(function (CustomFieldsGetRequest$) {
    /** @deprecated use `CustomFieldsGetRequest$inboundSchema` instead. */
    CustomFieldsGetRequest$.inboundSchema = exports.CustomFieldsGetRequest$inboundSchema;
    /** @deprecated use `CustomFieldsGetRequest$outboundSchema` instead. */
    CustomFieldsGetRequest$.outboundSchema = exports.CustomFieldsGetRequest$outboundSchema;
})(CustomFieldsGetRequest$ || (exports.CustomFieldsGetRequest$ = CustomFieldsGetRequest$ = {}));
function customFieldsGetRequestToJSON(customFieldsGetRequest) {
    return JSON.stringify(exports.CustomFieldsGetRequest$outboundSchema.parse(customFieldsGetRequest));
}
function customFieldsGetRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomFieldsGetRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomFieldsGetRequest' from JSON`);
}
