// @polar-sh/sdk@0.18.0/models/operations/discountsupdate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscountsUpdateRequest$ = exports.DiscountsUpdateRequest$outboundSchema = exports.DiscountsUpdateRequest$inboundSchema = void 0;
exports.discountsUpdateRequestToJSON = discountsUpdateRequestToJSON;
exports.discountsUpdateRequestFromJSON = discountsUpdateRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.DiscountsUpdateRequest$inboundSchema = z.object({
    id: z.string(),
    DiscountUpdate: components.DiscountUpdate$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "DiscountUpdate": "discountUpdate",
    });
});
/** @internal */
exports.DiscountsUpdateRequest$outboundSchema = z.object({
    id: z.string(),
    discountUpdate: components.DiscountUpdate$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        discountUpdate: "DiscountUpdate",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DiscountsUpdateRequest$;
(function (DiscountsUpdateRequest$) {
    /** @deprecated use `DiscountsUpdateRequest$inboundSchema` instead. */
    DiscountsUpdateRequest$.inboundSchema = exports.DiscountsUpdateRequest$inboundSchema;
    /** @deprecated use `DiscountsUpdateRequest$outboundSchema` instead. */
    DiscountsUpdateRequest$.outboundSchema = exports.DiscountsUpdateRequest$outboundSchema;
})(DiscountsUpdateRequest$ || (exports.DiscountsUpdateRequest$ = DiscountsUpdateRequest$ = {}));
function discountsUpdateRequestToJSON(discountsUpdateRequest) {
    return JSON.stringify(exports.DiscountsUpdateRequest$outboundSchema.parse(discountsUpdateRequest));
}
function discountsUpdateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DiscountsUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DiscountsUpdateRequest' from JSON`);
}
