// @polar-sh/sdk@0.18.0/models/operations/usersbenefitslist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsersBenefitsListResponse$ = exports.UsersBenefitsListResponse$outboundSchema = exports.UsersBenefitsListResponse$inboundSchema = exports.UsersBenefitsListRequest$ = exports.UsersBenefitsListRequest$outboundSchema = exports.UsersBenefitsListRequest$inboundSchema = exports.SubscriptionIDFilter$ = exports.SubscriptionIDFilter$outboundSchema = exports.SubscriptionIDFilter$inboundSchema = exports.OrderIDFilter$ = exports.OrderIDFilter$outboundSchema = exports.OrderIDFilter$inboundSchema = exports.OrganizationIDFilter$ = exports.OrganizationIDFilter$outboundSchema = exports.OrganizationIDFilter$inboundSchema = exports.BenefitTypeFilter$ = exports.BenefitTypeFilter$outboundSchema = exports.BenefitTypeFilter$inboundSchema = void 0;
exports.benefitTypeFilterToJSON = benefitTypeFilterToJSON;
exports.benefitTypeFilterFromJSON = benefitTypeFilterFromJSON;
exports.organizationIDFilterToJSON = organizationIDFilterToJSON;
exports.organizationIDFilterFromJSON = organizationIDFilterFromJSON;
exports.orderIDFilterToJSON = orderIDFilterToJSON;
exports.orderIDFilterFromJSON = orderIDFilterFromJSON;
exports.subscriptionIDFilterToJSON = subscriptionIDFilterToJSON;
exports.subscriptionIDFilterFromJSON = subscriptionIDFilterFromJSON;
exports.usersBenefitsListRequestToJSON = usersBenefitsListRequestToJSON;
exports.usersBenefitsListRequestFromJSON = usersBenefitsListRequestFromJSON;
exports.usersBenefitsListResponseToJSON = usersBenefitsListResponseToJSON;
exports.usersBenefitsListResponseFromJSON = usersBenefitsListResponseFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.BenefitTypeFilter$inboundSchema = z.union([
    components.BenefitType$inboundSchema,
    z.array(components.BenefitType$inboundSchema),
]);
/** @internal */
exports.BenefitTypeFilter$outboundSchema = z.union([
    components.BenefitType$outboundSchema,
    z.array(components.BenefitType$outboundSchema),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BenefitTypeFilter$;
(function (BenefitTypeFilter$) {
    /** @deprecated use `BenefitTypeFilter$inboundSchema` instead. */
    BenefitTypeFilter$.inboundSchema = exports.BenefitTypeFilter$inboundSchema;
    /** @deprecated use `BenefitTypeFilter$outboundSchema` instead. */
    BenefitTypeFilter$.outboundSchema = exports.BenefitTypeFilter$outboundSchema;
})(BenefitTypeFilter$ || (exports.BenefitTypeFilter$ = BenefitTypeFilter$ = {}));
function benefitTypeFilterToJSON(benefitTypeFilter) {
    return JSON.stringify(exports.BenefitTypeFilter$outboundSchema.parse(benefitTypeFilter));
}
function benefitTypeFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BenefitTypeFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BenefitTypeFilter' from JSON`);
}
/** @internal */
exports.OrganizationIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.OrganizationIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrganizationIDFilter$;
(function (OrganizationIDFilter$) {
    /** @deprecated use `OrganizationIDFilter$inboundSchema` instead. */
    OrganizationIDFilter$.inboundSchema = exports.OrganizationIDFilter$inboundSchema;
    /** @deprecated use `OrganizationIDFilter$outboundSchema` instead. */
    OrganizationIDFilter$.outboundSchema = exports.OrganizationIDFilter$outboundSchema;
})(OrganizationIDFilter$ || (exports.OrganizationIDFilter$ = OrganizationIDFilter$ = {}));
function organizationIDFilterToJSON(organizationIDFilter) {
    return JSON.stringify(exports.OrganizationIDFilter$outboundSchema.parse(organizationIDFilter));
}
function organizationIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrganizationIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrganizationIDFilter' from JSON`);
}
/** @internal */
exports.OrderIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.OrderIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderIDFilter$;
(function (OrderIDFilter$) {
    /** @deprecated use `OrderIDFilter$inboundSchema` instead. */
    OrderIDFilter$.inboundSchema = exports.OrderIDFilter$inboundSchema;
    /** @deprecated use `OrderIDFilter$outboundSchema` instead. */
    OrderIDFilter$.outboundSchema = exports.OrderIDFilter$outboundSchema;
})(OrderIDFilter$ || (exports.OrderIDFilter$ = OrderIDFilter$ = {}));
function orderIDFilterToJSON(orderIDFilter) {
    return JSON.stringify(exports.OrderIDFilter$outboundSchema.parse(orderIDFilter));
}
function orderIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrderIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrderIDFilter' from JSON`);
}
/** @internal */
exports.SubscriptionIDFilter$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
exports.SubscriptionIDFilter$outboundSchema = z.union([z.string(), z.array(z.string())]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var SubscriptionIDFilter$;
(function (SubscriptionIDFilter$) {
    /** @deprecated use `SubscriptionIDFilter$inboundSchema` instead. */
    SubscriptionIDFilter$.inboundSchema = exports.SubscriptionIDFilter$inboundSchema;
    /** @deprecated use `SubscriptionIDFilter$outboundSchema` instead. */
    SubscriptionIDFilter$.outboundSchema = exports.SubscriptionIDFilter$outboundSchema;
})(SubscriptionIDFilter$ || (exports.SubscriptionIDFilter$ = SubscriptionIDFilter$ = {}));
function subscriptionIDFilterToJSON(subscriptionIDFilter) {
    return JSON.stringify(exports.SubscriptionIDFilter$outboundSchema.parse(subscriptionIDFilter));
}
function subscriptionIDFilterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.SubscriptionIDFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubscriptionIDFilter' from JSON`);
}
/** @internal */
exports.UsersBenefitsListRequest$inboundSchema = z.object({
    type_filter: z.nullable(z.union([
        components.BenefitType$inboundSchema,
        z.array(components.BenefitType$inboundSchema),
    ])).optional(),
    organization_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    order_id: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    subscription_id: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.UserBenefitSortProperty$inboundSchema))
        .optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "type_filter": "typeFilter",
        "organization_id": "organizationId",
        "order_id": "orderId",
        "subscription_id": "subscriptionId",
    });
});
/** @internal */
exports.UsersBenefitsListRequest$outboundSchema = z.object({
    typeFilter: z.nullable(z.union([
        components.BenefitType$outboundSchema,
        z.array(components.BenefitType$outboundSchema),
    ])).optional(),
    organizationId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    orderId: z.nullable(z.union([z.string(), z.array(z.string())])).optional(),
    subscriptionId: z.nullable(z.union([z.string(), z.array(z.string())]))
        .optional(),
    page: z.number().int().default(1),
    limit: z.number().int().default(10),
    sorting: z.nullable(z.array(components.UserBenefitSortProperty$outboundSchema)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        typeFilter: "type_filter",
        organizationId: "organization_id",
        orderId: "order_id",
        subscriptionId: "subscription_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersBenefitsListRequest$;
(function (UsersBenefitsListRequest$) {
    /** @deprecated use `UsersBenefitsListRequest$inboundSchema` instead. */
    UsersBenefitsListRequest$.inboundSchema = exports.UsersBenefitsListRequest$inboundSchema;
    /** @deprecated use `UsersBenefitsListRequest$outboundSchema` instead. */
    UsersBenefitsListRequest$.outboundSchema = exports.UsersBenefitsListRequest$outboundSchema;
})(UsersBenefitsListRequest$ || (exports.UsersBenefitsListRequest$ = UsersBenefitsListRequest$ = {}));
function usersBenefitsListRequestToJSON(usersBenefitsListRequest) {
    return JSON.stringify(exports.UsersBenefitsListRequest$outboundSchema.parse(usersBenefitsListRequest));
}
function usersBenefitsListRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UsersBenefitsListRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UsersBenefitsListRequest' from JSON`);
}
/** @internal */
exports.UsersBenefitsListResponse$inboundSchema = z.object({
    Result: components.ListResourceUserBenefit$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "Result": "result",
    });
});
/** @internal */
exports.UsersBenefitsListResponse$outboundSchema = z.object({
    result: components.ListResourceUserBenefit$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        result: "Result",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UsersBenefitsListResponse$;
(function (UsersBenefitsListResponse$) {
    /** @deprecated use `UsersBenefitsListResponse$inboundSchema` instead. */
    UsersBenefitsListResponse$.inboundSchema = exports.UsersBenefitsListResponse$inboundSchema;
    /** @deprecated use `UsersBenefitsListResponse$outboundSchema` instead. */
    UsersBenefitsListResponse$.outboundSchema = exports.UsersBenefitsListResponse$outboundSchema;
})(UsersBenefitsListResponse$ || (exports.UsersBenefitsListResponse$ = UsersBenefitsListResponse$ = {}));
function usersBenefitsListResponseToJSON(usersBenefitsListResponse) {
    return JSON.stringify(exports.UsersBenefitsListResponse$outboundSchema.parse(usersBenefitsListResponse));
}
function usersBenefitsListResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UsersBenefitsListResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UsersBenefitsListResponse' from JSON`);
}
